# 1. 정신질환 게시글을 10개 이상인 사용자 리스트 수집

## 1) author list 만들기

In [2]:
import pandas as pd
import time
import datetime
import requests
import csv

In [7]:
df = pd.read_csv("anxiety\\anxiety.csv", names=["illness", "title", "author", "text_context", "time", "sub_id"])

In [8]:
df

illness                                              title  \
0       anxiety  I've never experienced anxiety this bad before...   
1       anxiety  MFW What I think is going to be a movie showin...   
2       anxiety                          4 year relationship ended   
3       anxiety                               A couple of seconds.   
4       anxiety  Everyone is the center of their experience, st...   
...         ...                                                ...   
298848  anxiety            Declined boss's call because of anxiety   
298849  anxiety  Does anyone know can you get like an online ps...   
298850  anxiety                               Hello I am panicking   
298851  anxiety                          UPDATE: She likes me back   
298852  anxiety              Need help with workout and relaxation   

                    author                                       text_context  \
0             coconutmango  22/F. I've been struggling with depression for...   
1               Nerdburton                                                NaN   
2             Starkiller32  My girlfriend of 4+ years and I just split up ...   
3                [deleted]                                          [deleted]   
4                [deleted]                                          [deleted]   
...                    ...                                                ...   
298848       Lost_Satelite  My anxiety eats me alive.\r\n\r\nEveryone is s...   
298849        Cedarxembers  I am curious if with everything happening now ...   
298850     throwaway_87653  Just an average Tuesday. Hypervigilant as fuck...   
298851  RandomGallonOfMilk  Just a text back from her and she likes me bac...   
298852             MOOWRLD  So my chest has been very heavy and tensed to ...   

                       time  sub_id  
0       2015-09-23 12:29:37  3m0ugg  
1       2015-09-23 12:49:49  3m0wp7  
2       2015-09-23 12:55:49  3m0xca  
3       2015-09-23 13:20:20  3m1019  
4       2015-09-23 13:44:39  3m12g6  
...                     ...     ...  
298848  2020-09-23 12:02:34  iy29fz  
298849  2020-09-23 12:03:11  iy29sy  
298850  2020-09-23 12:04:52  iy2are  
298851  2020-09-23 12:06:56  iy2bu4  
298852  2020-09-23 12:08:13  iy2ck1  

[298853 rows x 6 columns]

## 제거할 사항
1) df['author'] == '[deleted]'    
2) df['text_content'] == '[deleted]'    
3) df['text_content'] == '[removed]'    
2) df['text_content'].isnull()

결과: 60009 => 44734 (schizo_origin.csv)    
결과: 298853=>251990 (anxiety.csv)

In [9]:
deleted_author = df[df['author'] == '[deleted]']
df = df.drop(deleted_author.index)
df = df.drop(df[df['text_context'] == '[deleted]'].index)
df = df.drop(df[df['text_context'] == '[removed]'].index)
content_null = df[df['text_context'].isnull()].index
df= df.drop(content_null)

In [10]:
df

illness                                              title  \
0       anxiety  I've never experienced anxiety this bad before...   
2       anxiety                          4 year relationship ended   
5       anxiety  I cannot ask out a girl without feeling horrib...   
7       anxiety        Anxiety attacks at work. I found my trigger   
10      anxiety                           Constantly feeling sick.   
...         ...                                                ...   
298848  anxiety            Declined boss's call because of anxiety   
298849  anxiety  Does anyone know can you get like an online ps...   
298850  anxiety                               Hello I am panicking   
298851  anxiety                          UPDATE: She likes me back   
298852  anxiety              Need help with workout and relaxation   

                    author                                       text_context  \
0             coconutmango  22/F. I've been struggling with depression for...   
2             Starkiller32  My girlfriend of 4+ years and I just split up ...   
5               ohnobarnes  I understand how difficult and awkward it can ...   
7           Captaine_Cooke  Hey All\r\n\r\nThought I'd share my story and ...   
10                 mikej90  For a little more than a year I've been feelin...   
...                    ...                                                ...   
298848       Lost_Satelite  My anxiety eats me alive.\r\n\r\nEveryone is s...   
298849        Cedarxembers  I am curious if with everything happening now ...   
298850     throwaway_87653  Just an average Tuesday. Hypervigilant as fuck...   
298851  RandomGallonOfMilk  Just a text back from her and she likes me bac...   
298852             MOOWRLD  So my chest has been very heavy and tensed to ...   

                       time  sub_id  
0       2015-09-23 12:29:37  3m0ugg  
2       2015-09-23 12:55:49  3m0xca  
5       2015-09-23 13:45:12  3m12ik  
7       2015-09-23 15:13:25  3m1ac2  
10      2015-09-23 17:03:38  3m1iw6  
...                     ...     ...  
298848  2020-09-23 12:02:34  iy29fz  
298849  2020-09-23 12:03:11  iy29sy  
298850  2020-09-23 12:04:52  iy2are  
298851  2020-09-23 12:06:56  iy2bu4  
298852  2020-09-23 12:08:13  iy2ck1  

[251990 rows x 6 columns]

## 제거할 사항2
1) 사용자별로 groupby 진행    
2) 사용자별 게시글 갯수가 10개 이상인 사용자만 남기기

결과: 14534 => 744 (schizo_origin.csv)    
결과: 139444 => 2143 (anxiety.csv)

In [11]:
# 사용자별 게시글 갯수 확인을 위한 grouping
author_grouped = df.groupby(by=['author'], as_index=False).count()
author_grouped

author  illness  title  text_context  time  sub_id
0            --------uuhhhhh        1      1             1     1       1
1                  ----0____        1      1             1     1       1
2                   ----fake        1      1             1     1       1
3       ---TheFierceDeity---        2      2             2     2       2
4                 ---okay---        1      1             1     1       1
...                      ...      ...    ...           ...   ...     ...
139439       zzzsadwinnerzzz        1      1             1     1       1
139440            zzzsss1233        1      1             1     1       1
139441            zzzz_on_me        1      1             1     1       1
139442            zzzzpuzzle        1      1             1     1       1
139443           zzzzzzzzzzd        1      1             1     1       1

[139444 rows x 6 columns]

In [12]:
ten_author = author_grouped[author_grouped['text_context'] > 9]
ten_author

author  illness  title  text_context  time  sub_id
139            -Skelan-       16     16            16    16      16
196     -bannedaccount-       12     12            12    12      12
541       1000ancestors       12     12            12    12      12
575     104degrees_cold       12     12            12    12      12
677      1234swkisgar56       16     16            16    16      16
...                 ...      ...    ...           ...   ...     ...
139049          zelis42       44     44            44    44      44
139073          zensama       16     16            16    16      16
139227    zmicromillion       17     17            17    17      17
139293    zombiemuss106       23     23            23    23      23
139303       zonbikoume       10     10            10    10      10

[2143 rows x 6 columns]

### 최종 author_list

In [13]:
author_list = ten_author['author']

## 2) author별 최초작성일 수집

In [14]:
import json
def getPushshiftData(after, before, sub, author):
    url = 'https://api.pushshift.io/reddit/search/submission/?size=1000&after='+str(after)+'&before='+str(before)+'&subreddit='+str(sub)+'&author='+str(author)
    r = requests.get(url)
    data = json.loads(r.text)
    return data['data']

In [15]:
def collectSubData(subm): # 글전체 갖고오기 ->이거 하는게 제일 낫긴 함
    subData = list() #list to store data points
    title = subm['title']
    try:
        selftext = subm['selftext']
    except:
        selftext = 'none'
    finally:
        subreddit = subm['subreddit']
        author = subm['author']
        sub_id = subm['id']
        if(subreddit == "anxiety"): ##################병명 변경#############################
            subredddit_check = 1
        else:
            subredddit_check = 0
        created = datetime.datetime.fromtimestamp(subm['created_utc'])
        subData.append((subreddit,author,created))
        subStats[sub_id] = subData

In [16]:
start = time.time()
error_author = [] 
first_date = []
illness = "anxiety"  ##################병명 변경#############################
for author in author_list:
    subStats = {}
    try:
        after = 1119604143
        before = 1600831376
        # 파일 수집
        subCount = 0
        data = getPushshiftData(after, before, illness, author)
        # Will run until all posts have been gathered 
        # from the 'after' date up until before date
        while len(data) > 0:
            for submission in data:
                collectSubData(submission)
                subCount+=1
            # Calls getPushshiftData() with the created date of the last submission
            print(len(data), " : ", author)
            print(str(datetime.datetime.fromtimestamp(data[-1]['created_utc'])))
            break
    except:
        error_author.append(author)
        print(error_author)
    # 파일 저장
    f = open("anxiety\\author_firstdate_anxiety.csv",'a',encoding='utf-8-sig') ##################병명 변경#############################
    wr = csv.writer(f)
    wr.writerow(['subreddit', 'author', "first_date"])
    for i in list(subStats.values()):
        wr.writerow(i[0])
        break
    f.close()

16  :  -Skelan-
2020-09-22 07:23:24
13  :  -bannedaccount-
2020-05-21 19:44:45
19  :  1000ancestors
2020-02-16 07:27:29
20  :  104degrees_cold
2020-06-15 07:02:44
18  :  1234swkisgar56
2020-08-12 22:20:54
28  :  123anxiety
2017-02-22 02:30:13
12  :  123grimey
2020-01-09 00:39:49
26  :  123likeabirdimfree
2020-04-18 00:11:19
25  :  1398_Days
2019-12-30 09:52:00
37  :  13yearoldidiot
2019-07-11 11:16:39
12  :  1dude090834
2018-09-03 09:27:31
10  :  1u1y
2019-08-03 04:03:23
14  :  2012lurk
2019-04-14 16:05:35
10  :  204coffee
2019-08-17 02:02:57
12  :  2621759912014199
2019-02-19 08:42:08
15  :  26_spanahh
2018-10-03 09:30:46
25  :  2906BC
2019-07-22 04:05:29
13  :  449_user
2019-03-28 08:07:09
17  :  449user
2018-02-14 11:55:12
11  :  48457601
2017-09-10 03:09:28
10  :  5021NJnj
2020-09-05 09:36:01
21  :  5280MikeyB
2020-07-11 18:29:44
12  :  553684
2020-06-06 21:46:22
49  :  5secondstozerotime
2019-07-21 04:16:52
27  :  7ammy30
2019-08-24 08:30:11
19  :  89SC
2020-09-08 21:26:43
13  :  

14  :  Csalz94009
2020-05-10 03:41:09
15  :  CuckoldCredenza
2018-10-22 05:30:40
14  :  Cupcakesalways
2017-06-29 08:10:23
14  :  Curlaub
2018-01-20 23:50:12
12  :  Cute-Squishy
2020-07-04 12:40:15
10  :  Cwreck92
2020-09-13 12:05:54
25  :  CyberWolfMVP
2020-06-29 12:11:10
30  :  DCT1997
2020-07-14 08:26:36
27  :  DIRj67
2019-01-21 15:03:45
11  :  DJT_MAGA_2020g
2018-12-14 02:28:26
14  :  DLuX94
2020-06-08 15:11:03
15  :  DXD4226
2020-06-01 20:51:38
22  :  DaaaamnOkay
2020-07-17 00:25:16
10  :  Dalek1234
2017-08-15 16:58:55
11  :  Damianjmcgrath
2018-06-13 07:00:52
11  :  DancingDiddy
2020-09-15 13:14:07
27  :  DaniDaniBoBani93
2018-09-10 07:46:48
11  :  Danoodlepod
2018-12-14 06:20:58
26  :  DarkCelestial
2020-08-07 12:01:35
15  :  DarkSkullMango
2018-12-04 09:27:15
38  :  Dark_and_gothic
2016-07-08 14:42:01
15  :  Darkencypher
2018-10-26 22:50:10
20  :  Daxter999
2020-07-14 14:01:55
19  :  DeadGravityyy
2017-08-15 12:08:05
13  :  DeadShadowKitty
2019-12-26 06:01:11
11  :  Deaddrop88


21  :  J-dude720
2020-02-11 15:19:06
16  :  J1mcdonald
2019-07-06 13:04:56
12  :  JD_Revan451
2019-03-22 00:34:44
38  :  JJ32Moose18
2019-03-18 11:37:04
19  :  JPalmer1992
2019-06-10 02:31:10
12  :  JTAnimations
2020-05-03 07:05:10
12  :  Jackson530
2020-07-11 15:58:11
14  :  JadedGypsy2238
2020-07-24 07:46:48
12  :  JaeKing2k
2019-03-15 20:28:20
14  :  JakeFromStateFarm100
2019-12-28 14:35:49
12  :  JamieOvechkin
2019-01-19 05:54:38
23  :  JasonRedux
2019-09-05 23:08:48
56  :  Jasonwj322a
2020-05-19 23:57:46
11  :  Jaydeisthename
2018-10-13 10:37:16
12  :  Jazzcabbageee
2020-06-29 08:06:45
19  :  Jburggss
2020-09-05 05:03:23
14  :  Jefftommens
2018-06-29 06:12:35
13  :  Jennyvex
2018-10-27 23:28:43
12  :  Jesta23
2017-10-29 04:33:54
23  :  JesusHMontgomery
2017-05-02 23:19:48
11  :  Jhp34
2016-12-07 09:01:36
23  :  Joe_Sizer
2020-08-22 20:53:15
17  :  JohnnyLong123
2019-03-03 03:33:17
13  :  Jonno182
2017-07-08 05:00:40
60  :  JonnyTsunami12
2019-11-16 14:03:33
34  :  JordanLeigh7
202

100  :  ODMAN03
2019-04-16 05:22:54
20  :  ON1YFRANK
2019-06-22 22:58:16
11  :  ObliviateProtego
2020-02-01 03:40:18
100  :  ObscuredByCIouds
2016-12-21 08:39:34
46  :  ObviousThrowaway4556
2018-06-27 15:00:28
57  :  Ocelot859
2020-09-21 03:58:51
23  :  OddK1dNorth
2020-06-25 00:07:52
21  :  Odysseus6
2020-08-22 09:13:53
16  :  OhMaiMai1993
2020-09-04 21:31:58
17  :  OhSheGlows
2020-01-28 00:39:59
11  :  Oh_my_Japanese_Boy
2020-02-29 00:16:13
49  :  OkSundae5
2020-07-18 08:07:22
25  :  OkText00
2020-06-11 07:57:50
12  :  Ok_Radish9966
2020-09-22 12:40:05
29  :  OldBren
2019-12-28 01:09:55
17  :  Old_boy7
2020-06-13 09:39:32
74  :  OliveEyes-
2017-02-14 10:36:21
14  :  Oliwine
2020-02-15 23:21:56
42  :  OneDayIllOwnALambo
2019-03-11 14:34:41
12  :  OneHotHamSandwich
2020-01-26 02:43:42
19  :  One_day-at-a_time
2019-10-19 14:52:05
22  :  Ooozu
2019-12-16 08:08:11
44  :  OpAgents
2020-03-06 03:43:30
18  :  Opppppe
2020-08-23 09:45:53
14  :  OptimalDragonflies
2020-03-24 22:49:03
18  :  Or

13  :  TWJ32
2019-10-29 05:17:37
19  :  TX_2_GA
2018-08-28 00:44:37
14  :  T_Eighteen
2020-07-08 12:44:59
28  :  TallQueer9
2017-01-02 13:07:30
28  :  TaraBURGER
2020-09-09 21:53:03
18  :  TaurineLine719
2018-11-05 19:34:51
61  :  Taylor6534
2019-08-16 06:31:32
11  :  Tbirdfan190
2020-07-27 10:25:40
13  :  TeezusRa
2018-02-21 22:46:39
18  :  TellMeWhatToDoDoctor
2019-10-03 22:16:16
18  :  TexanLoneStar
2020-09-17 09:14:11
15  :  That-Kid-Az
2020-08-06 12:54:50
31  :  ThatGuyNoOneRemember
2020-03-27 09:28:40
40  :  ThatOneNiggaFromMars
2018-10-19 04:49:32
14  :  Thatonemexicanchick
2018-08-22 02:43:38
10  :  TheAmazingPikachu
2019-07-06 22:34:16
10  :  TheAnxiousAsshole
2016-03-26 09:56:52
13  :  TheButtholeOfBravery
2019-04-01 07:33:28
28  :  TheColorOfSnails
2020-04-03 16:43:29
21  :  TheCynicPress
2020-07-02 06:24:25
19  :  TheElfishOne
2019-06-26 23:24:34
19  :  TheFakeZzig
2020-05-14 01:38:48
16  :  TheGreatMclovin
2018-10-26 01:37:44
23  :  TheGutenbergMachine
2019-08-27 06:36:20


11  :  afzalhai
2020-08-27 00:09:13
17  :  ageekyninja
2017-11-29 21:26:24
11  :  agrokrang
2018-04-20 21:10:08
10  :  ahartzer2015
2018-05-01 08:44:51
11  :  ahrahtnamas
2018-06-09 09:42:19
12  :  aidenaizumi
2020-08-17 13:14:26
27  :  ajc772
2020-05-05 02:46:45
22  :  ajdctx12
2020-01-17 06:40:32
28  :  ajdctx13
2019-11-17 08:39:19
18  :  ajk999
2019-11-17 04:36:36
53  :  akahaus
2019-02-25 03:04:21
47  :  alakazamSam
2020-08-23 09:20:47
38  :  albert2749
2020-07-12 19:27:32
26  :  alessF
2020-04-05 13:06:44
16  :  alexmacmusic
2020-07-28 02:50:42
15  :  alexmagggggs
2020-09-09 19:35:40
15  :  alittleloopy17
2020-07-10 16:08:42
28  :  alkatori
2017-08-02 08:07:16
17  :  allicat4
2019-03-31 04:00:58
14  :  allprotravis
2016-01-13 07:17:30
79  :  allthekos
2020-08-17 03:01:03
18  :  alltheothersrtaken
2017-04-25 16:22:22
15  :  almondmilkicedchai
2019-11-08 13:18:05
14  :  aloe77
2019-12-29 23:58:06
11  :  aloevera81
2020-08-05 23:58:39
91  :  alphafighter09
2020-01-14 16:09:33
17  :  

41  :  coffeegreentea
2018-06-01 14:24:25
11  :  cokesyrup
2018-12-30 12:07:06
26  :  colexthoughts
2019-11-15 06:35:11
12  :  comfortablynumb5
2019-04-04 13:07:27
23  :  comicgrl211
2018-02-26 01:36:17
12  :  comingheretocry
2020-09-22 17:07:51
12  :  completelybasic
2019-07-08 06:31:10
16  :  confused123456
2019-09-27 01:24:28
14  :  confusedteen12
2020-05-18 06:42:51
20  :  confusing_rants
2020-09-17 05:37:46
12  :  constantheadpain
2020-01-09 17:26:45
23  :  cookofthesea
2019-03-12 09:15:09
11  :  coolfishes
2020-04-17 16:38:08
19  :  coolman1028
2020-02-29 05:53:52
11  :  cory667
2019-11-18 22:32:27
10  :  couldashoudlamassa
2020-04-15 22:57:36
41  :  courtneat
2020-09-11 23:31:23
11  :  creampuff313
2019-02-06 10:05:00
11  :  creepy_pie31
2020-01-31 06:16:11
17  :  crexyisaword
2019-06-19 12:21:06
10  :  cripple2493
2020-01-30 12:08:14
11  :  csb7566381
2020-09-15 07:44:14
17  :  cultmovieaddict
2019-05-18 09:22:08
11  :  cupidshufflechaslide
2020-03-31 04:36:35
24  :  cyberv123


10  :  hylianhylia
2019-10-20 11:02:35
11  :  iHateNumbers123
2020-05-30 22:35:33
14  :  ia1v1chem
2020-03-31 02:02:20
12  :  iamcorneliusNL
2018-12-05 21:55:59
15  :  icarly1234
2020-09-11 05:24:33
10  :  ice_cold_sandwich
2020-05-13 06:41:53
15  :  iconiqe
2020-03-29 21:08:29
10  :  idontevenknowrandom
2020-09-12 01:00:05
10  :  igotitatriteaid
2020-09-03 14:55:27
33  :  ihatelifexoxox
2020-09-22 06:30:09
12  :  ilikebatman69
2020-02-21 00:06:15
13  :  illstayanonymous
2019-06-06 00:34:21
13  :  imGoingToEatYourTots
2020-08-10 02:33:33
16  :  imamess777
2020-08-18 13:41:40
64  :  imcrazyfordisone
2019-09-30 05:17:52
10  :  immh2001
2020-02-06 00:22:53
37  :  indifferences
2019-10-18 14:53:48
12  :  ineedtoknowmorenow
2019-03-16 06:18:04
18  :  infantry_of_the_line
2018-08-16 23:51:20
11  :  infinitytwat
2020-05-30 23:49:56
12  :  intergalacticpup
2020-07-20 10:34:44
13  :  irish-boyyaya
2019-04-17 06:50:57
14  :  isaonreddit
2018-12-05 03:18:31
13  :  itdoesntmatterbabes
2020-08-25 1

17  :  mimirabbit
2020-09-07 19:50:25
24  :  mink2018
2020-07-16 16:14:27
14  :  minseokie23
2018-12-28 13:16:37
15  :  miscaccounts
2020-06-26 14:49:55
10  :  miscarriagex2
2020-05-06 11:14:36
13  :  mishkabear22
2020-04-15 19:36:33
13  :  mismashful
2020-05-04 03:55:12
12  :  misosoup888
2020-08-30 10:35:01
13  :  mista_steam_ya_hams
2019-01-11 22:30:43
15  :  mitronaguay
2019-10-03 05:24:17
59  :  mjpenslitbooksgalore
2020-09-11 06:20:12
15  :  mkh4990
2020-04-12 04:13:56
40  :  mmcca
2020-05-08 02:06:49
10  :  moake740
2019-04-22 05:25:51
30  :  modernjaneausten
2020-09-14 15:32:36
17  :  mojomonkey1
2018-07-15 10:04:06
11  :  moldyskeleton
2020-07-08 08:57:46
20  :  momekeqkupto-
2020-07-07 03:37:35
10  :  momluvr
2017-10-16 03:29:46
14  :  monsterrat88
2018-11-15 06:20:15
18  :  morgankr34
2019-12-08 12:25:14
23  :  morimushroom
2020-09-22 02:29:26
41  :  mostlikelytocry
2020-09-20 21:40:07
12  :  movimento9
2018-03-28 02:26:41
11  :  mraeons12345
2019-01-01 15:54:36
29  :  mrbos

16  :  saltyman420
2019-04-06 05:20:39
14  :  sam8998
2020-09-20 02:50:00
15  :  sambamsatan
2020-01-22 04:45:59
12  :  samema
2019-10-26 23:36:17
14  :  samgutz99
2018-10-09 07:54:50
51  :  sammyjamez
2019-04-01 03:31:41
13  :  sanalang
2017-08-16 07:18:43
51  :  sangbum60090
2020-09-18 18:29:06
17  :  satancookie22
2019-11-25 15:00:31
17  :  sataneatstomatoes
2020-08-06 02:09:41
23  :  saturnparty
2020-09-23 07:44:19
15  :  saultarus
2020-09-21 01:16:31
12  :  scarpmclovin
2019-08-27 03:20:32
21  :  sceerdycat
2020-04-10 22:22:26
16  :  sco8881
2020-06-20 06:23:09
21  :  scurtel
2020-08-07 07:45:01
16  :  seahunter1815
2018-08-21 02:37:28
18  :  sean7755
2020-02-03 15:06:21
12  :  seekingstars
2020-07-12 00:40:10
22  :  seniilisiili
2020-09-15 00:19:15
14  :  serendipity1996
2019-02-21 03:27:19
20  :  seven2812
2019-08-15 10:29:23
10  :  sflo97
2020-09-01 10:27:01
11  :  shadowgun190
2017-08-31 05:54:29
14  :  shadowofassassin
2020-08-29 13:16:12
30  :  shakespearestark
2020-09-17 07

11  :  vaasan
2020-07-09 21:55:37
27  :  valentinagd01
2020-08-03 12:45:10
10  :  valueyoghurt
2020-04-15 08:05:05
17  :  vanessawinters
2019-12-15 05:07:14
36  :  veganburritokiller
2020-03-28 01:01:18
12  :  veggietown4
2020-06-03 04:28:34
22  :  veraloe
2019-05-15 11:53:50
13  :  veryberryblue
2020-05-31 10:48:28
11  :  victorsledge07
2017-06-03 20:49:59
15  :  vietusernam
2019-03-30 10:47:00
32  :  vincchensko-123
2019-04-24 18:11:37
59  :  vitalogy95
2020-03-01 20:45:22
10  :  volunteerdoorknob
2019-04-24 10:07:06
17  :  vorpalJ
2020-03-12 04:45:34
10  :  vsvpflex
2019-11-04 14:17:25
22  :  vulepsy
2020-07-18 12:35:22
10  :  wagalow
2020-01-29 08:11:03
13  :  wahoo000
2020-05-18 12:32:08
12  :  wannabENID
2020-01-16 21:01:24
17  :  wasabicupcakes
2019-01-19 07:30:31
13  :  wastedtime666
2019-05-15 09:37:16
11  :  waterpanne
2020-04-06 14:00:46
10  :  wateryeyes97
2020-08-01 21:09:52
16  :  waves-weeb
2020-03-30 15:07:55
13  :  weedquestion123
2019-08-16 06:17:18
17  :  weegee12345

## 3) author 별 게시글 수집

reddit 정신질환 최초작성일 이전의 2년간 다른 하위레딧의 작성글 수집

In [8]:
import json
def getPushshiftData(after, before, author):
    url = 'https://api.pushshift.io/reddit/search/submission/?size=1000&after='+str(after)+'&before='+str(before)+'&author='+str(author)
    r = requests.get(url)
    data = json.loads(r.text)
    return data['data']

In [9]:
def collectSubData(subm): # 글전체 갖고오기 ->이거 하는게 제일 낫긴 함
    subData = list() #list to store data points
    title = subm['title']
    try:
        selftext = subm['selftext']
    except:
        selftext = 'none'
    finally:
        subreddit = subm['subreddit']
        author = subm['author']
        sub_id = subm['id']
        if(subreddit == "anxiety"): ##################병명 변경#############################
            subredddit_check = 1
        else:
            subredddit_check = 0
        created = datetime.datetime.fromtimestamp(subm['created_utc'])
        subData.append((subreddit,title,author,selftext,created, subredddit_check))
        subStats[sub_id] = subData

In [10]:
# 사용자별 최초작성일 저장 csv 읽어오기
author_df = pd.read_csv("anxiety\\author_firstdate_anxiety.csv")

In [11]:
author_df

subreddit           author           first_date
0      Anxiety         -Skelan-  2019-12-29 06:50:15
1      Anxiety  -bannedaccount-  2019-07-09 01:22:03
2      Anxiety    1000ancestors  2014-03-14 05:48:16
3      Anxiety  104degrees_cold  2020-05-27 11:28:40
4      Anxiety   1234swkisgar56  2018-01-30 11:12:39
...        ...              ...                  ...
2138   Anxiety          zelis42  2017-12-09 21:11:18
2139   Anxiety          zensama  2015-04-26 14:17:38
2140   Anxiety    zmicromillion  2017-01-28 02:51:03
2141   Anxiety    zombiemuss106  2019-12-29 16:57:20
2142   Anxiety       zonbikoume  2017-11-11 17:08:37

[2143 rows x 3 columns]

In [1]:
# 혹시 한줄씩 띄어서 저장되는 경우 처리
author_df = author_df.drop(author_df[author_df['author'] == 'author'].index)
author_df.to_csv("anxiety\\author_firstdate_anxiety.csv", index = False)

NameError: name 'author_df' is not defined

In [1]:
def convert_unixtime(date_time):
    """Convert datetime to unixtime"""
    import datetime
    unixtime = datetime.datetime.strptime(date_time,
                               '%Y-%m-%d %H:%M:%S').timestamp()
    return unixtime


In [13]:
start = time.time()
error_author = [] 
#가끔 에러가 나는 케이스도 있어서 일단 별도로 저장해두는 리스트입니당
subStats = {}
for i in range(len(author_df)):
    author = author_df['author'].iloc[i]
    try:
        after = int(convert_unixtime(author_df['first_date'].iloc[i])) - 63158400
        before = author_df['first_date'].iloc[i]
        # 파일 수집
        subCount = 0
        data = getPushshiftData(after, before, author)
        # Will run until all posts have been gathered 
        # from the 'after' date up until before date
        while len(data) > 0:
            for submission in data:
                collectSubData(submission)
                subCount+=1
            # Calls getPushshiftData() with the created date of the last submission
            print(len(data), " : ", author)
            print(str(datetime.datetime.fromtimestamp(data[-1]['created_utc'])))
            after = data[-1]['created_utc']
            while True:
                try:
                    data = getPushshiftData(after, before, author)
                    break
                except:
                    print('에러 발생')
                    continue
    except:
        error_author.append(author)
        print(error_author)

                    
# 파일 저장
f = open("anxiety\\anxiety_author_before.csv",'a',encoding='utf-8-sig') ##################병명 변경#############################
wr = csv.writer(f)
wr.writerow(['subreddit', 'title', 'author', 'selftext', "date", "subredddit_check"])
for i in list(subStats.values()):
    wr.writerow(i[0])
f.close()
print(author, " 파일 저장 성공!")
print('걸린시간:'+ str(time.time()-start))

1  :  -Skelan-
2019-12-29 06:50:15
7  :  -bannedaccount-
2019-07-09 01:22:03
4  :  1000ancestors
2014-03-14 05:48:16
18  :  104degrees_cold
2020-05-27 11:28:40
12  :  1234swkisgar56
2018-01-30 11:12:39
4  :  123anxiety
2015-02-20 10:18:22
17  :  123grimey
2017-09-10 21:18:14
11  :  123likeabirdimfree
2015-12-12 05:50:06
1  :  1398_Days
2016-03-28 05:13:04
90  :  13yearoldidiot
2019-06-04 09:58:14
1  :  1dude090834
2018-08-02 11:16:47
2  :  1u1y
2019-05-17 01:20:16
12  :  2012lurk
2016-05-01 09:51:09
10  :  204coffee
2018-11-05 01:51:14
1  :  2621759912014199
2017-04-17 01:42:28
100  :  26_spanahh
2018-02-16 12:11:23
86  :  26_spanahh
2018-03-18 11:43:56
7  :  2906BC
2017-08-11 18:56:15
1  :  449_user
2018-10-17 04:43:31
1  :  449user
2017-05-09 03:19:10
16  :  48457601
2016-08-28 05:45:53
1  :  5021NJnj
2020-02-22 21:27:48
19  :  5280MikeyB
2020-03-18 09:28:04
3  :  553684
2019-09-08 21:14:16
100  :  5secondstozerotime
2014-04-13 15:04:39
45  :  5secondstozerotime
2014-09-04 04:07:20
3

100  :  AutoModerator
2014-07-02 19:02:25
100  :  AutoModerator
2014-07-03 22:02:16
100  :  AutoModerator
2014-07-04 22:03:03
100  :  AutoModerator
2014-07-06 02:04:15
100  :  AutoModerator
2014-07-07 07:02:18
100  :  AutoModerator
2014-07-08 02:02:09
100  :  AutoModerator
2014-07-09 09:46:09
100  :  AutoModerator
2014-07-10 10:02:09
100  :  AutoModerator
2014-07-11 19:02:19
100  :  AutoModerator
2014-07-12 19:02:32
100  :  AutoModerator
2014-07-13 22:02:30
100  :  AutoModerator
2014-07-14 21:02:39
100  :  AutoModerator
2014-07-15 21:02:08
100  :  AutoModerator
2014-07-16 21:32:11
100  :  AutoModerator
2014-07-17 23:02:42
100  :  AutoModerator
2014-07-18 22:30:08
100  :  AutoModerator
2014-07-20 01:02:12
100  :  AutoModerator
2014-07-21 04:00:11
100  :  AutoModerator
2014-07-21 22:03:35
100  :  AutoModerator
2014-07-22 22:02:50
100  :  AutoModerator
2014-07-23 23:00:08
100  :  AutoModerator
2014-07-25 01:02:30
100  :  AutoModerator
2014-07-26 01:18:12
100  :  AutoModerator
2014-07-27 0

100  :  AutoModerator
2014-10-29 23:02:14
100  :  AutoModerator
2014-10-30 16:02:15
100  :  AutoModerator
2014-10-31 02:28:09
100  :  AutoModerator
2014-10-31 19:02:19
100  :  AutoModerator
2014-11-01 01:02:17
100  :  AutoModerator
2014-11-01 16:22:12
100  :  AutoModerator
2014-11-02 01:02:26
100  :  AutoModerator
2014-11-02 19:02:19
100  :  AutoModerator
2014-11-03 04:28:09
100  :  AutoModerator
2014-11-03 17:02:14
100  :  AutoModerator
2014-11-03 22:06:14
100  :  AutoModerator
2014-11-04 11:26:08
100  :  AutoModerator
2014-11-04 23:02:07
100  :  AutoModerator
2014-11-05 14:28:09
100  :  AutoModerator
2014-11-05 22:02:23
100  :  AutoModerator
2014-11-06 16:02:24
100  :  AutoModerator
2014-11-07 02:00:23
100  :  AutoModerator
2014-11-07 18:28:10
100  :  AutoModerator
2014-11-08 00:02:43
100  :  AutoModerator
2014-11-08 16:18:14
100  :  AutoModerator
2014-11-09 03:28:12
100  :  AutoModerator
2014-11-09 19:02:21
100  :  AutoModerator
2014-11-10 05:52:16
100  :  AutoModerator
2014-11-10 1

100  :  AutoModerator
2015-01-27 05:34:41
100  :  AutoModerator
2015-01-27 17:22:18
100  :  AutoModerator
2015-01-28 00:02:27
100  :  AutoModerator
2015-01-28 14:02:23
100  :  AutoModerator
2015-01-28 21:00:37
100  :  AutoModerator
2015-01-29 05:02:19
100  :  AutoModerator
2015-01-29 19:02:17
100  :  AutoModerator
2015-01-30 01:02:25
100  :  AutoModerator
2015-01-30 14:28:10
100  :  AutoModerator
2015-01-30 21:02:18
100  :  AutoModerator
2015-01-31 02:28:13
100  :  AutoModerator
2015-01-31 15:02:15
100  :  AutoModerator
2015-01-31 23:02:17
100  :  AutoModerator
2015-02-01 09:28:11
100  :  AutoModerator
2015-02-01 19:02:19
100  :  AutoModerator
2015-02-02 00:02:23
100  :  AutoModerator
2015-02-02 09:28:27
100  :  AutoModerator
2015-02-02 18:02:17
100  :  AutoModerator
2015-02-02 22:02:57
100  :  AutoModerator
2015-02-03 04:32:21
100  :  AutoModerator
2015-02-03 17:16:11
100  :  AutoModerator
2015-02-04 00:02:20
100  :  AutoModerator
2015-02-04 10:32:32
100  :  AutoModerator
2015-02-04 2

100  :  AutoModerator
2015-04-08 02:30:23
100  :  AutoModerator
2015-04-08 16:00:23
100  :  AutoModerator
2015-04-08 21:48:20
100  :  AutoModerator
2015-04-09 03:48:19
100  :  AutoModerator
2015-04-09 17:02:16
100  :  AutoModerator
2015-04-09 22:10:21
100  :  AutoModerator
2015-04-10 09:00:25
100  :  AutoModerator
2015-04-10 18:02:09
100  :  AutoModerator
2015-04-10 22:00:21
100  :  AutoModerator
2015-04-11 03:02:11
100  :  AutoModerator
2015-04-11 14:02:20
100  :  AutoModerator
2015-04-11 21:00:31
100  :  AutoModerator
2015-04-12 06:02:13
100  :  AutoModerator
2015-04-12 17:02:09
100  :  AutoModerator
2015-04-12 22:02:23
100  :  AutoModerator
2015-04-13 07:32:09
100  :  AutoModerator
2015-04-13 15:02:21
100  :  AutoModerator
2015-04-13 20:32:10
100  :  AutoModerator
2015-04-13 23:02:27
100  :  AutoModerator
2015-04-14 09:02:05
100  :  AutoModerator
2015-04-14 19:02:17
100  :  AutoModerator
2015-04-14 23:02:35
100  :  AutoModerator
2015-04-15 12:00:24
100  :  AutoModerator
2015-04-15 2

100  :  AutoModerator
2015-06-06 09:02:08
100  :  AutoModerator
2015-06-06 17:02:22
100  :  AutoModerator
2015-06-06 22:42:09
100  :  AutoModerator
2015-06-07 09:02:14
100  :  AutoModerator
2015-06-07 17:02:13
100  :  AutoModerator
2015-06-07 22:02:20
100  :  AutoModerator
2015-06-08 07:02:05
100  :  AutoModerator
2015-06-08 12:56:08
100  :  AutoModerator
2015-06-08 18:02:08
100  :  AutoModerator
2015-06-08 21:02:39
100  :  AutoModerator
2015-06-09 01:02:17
100  :  AutoModerator
2015-06-09 11:12:17
100  :  AutoModerator
2015-06-09 19:02:37
100  :  AutoModerator
2015-06-09 23:22:06
100  :  AutoModerator
2015-06-10 10:02:07
100  :  AutoModerator
2015-06-10 18:48:15
100  :  AutoModerator
2015-06-10 22:06:09
100  :  AutoModerator
2015-06-11 06:48:15
100  :  AutoModerator
2015-06-11 17:02:09
100  :  AutoModerator
2015-06-11 22:02:17
100  :  AutoModerator
2015-06-12 07:02:05
100  :  AutoModerator
2015-06-12 15:02:18
100  :  AutoModerator
2015-06-12 20:02:04
100  :  AutoModerator
2015-06-12 2

100  :  AutoModerator
2015-07-29 04:32:17
100  :  AutoModerator
2015-07-29 14:02:28
100  :  AutoModerator
2015-07-29 20:02:17
100  :  AutoModerator
2015-07-29 23:02:16
100  :  AutoModerator
2015-07-30 08:02:07
100  :  AutoModerator
2015-07-30 17:02:20
100  :  AutoModerator
2015-07-30 21:32:12
100  :  AutoModerator
2015-07-31 03:02:37
100  :  AutoModerator
2015-07-31 13:02:14
100  :  AutoModerator
2015-07-31 18:02:28
100  :  AutoModerator
2015-07-31 21:02:07
100  :  AutoModerator
2015-08-01 00:02:25
100  :  AutoModerator
2015-08-01 07:02:32
100  :  AutoModerator
2015-08-01 14:02:38
100  :  AutoModerator
2015-08-01 19:00:45
100  :  AutoModerator
2015-08-01 23:00:20
100  :  AutoModerator
2015-08-02 04:00:26
100  :  AutoModerator
2015-08-02 14:02:28
100  :  AutoModerator
2015-08-02 19:02:25
100  :  AutoModerator
2015-08-02 23:02:28
100  :  AutoModerator
2015-08-03 08:30:13
100  :  AutoModerator
2015-08-03 13:06:27
100  :  AutoModerator
2015-08-03 17:56:15
100  :  AutoModerator
2015-08-03 2

100  :  AutoModerator
2015-09-14 18:03:20
100  :  AutoModerator
2015-09-14 21:06:27
100  :  AutoModerator
2015-09-15 00:03:03
100  :  AutoModerator
2015-09-15 06:32:12
100  :  AutoModerator
2015-09-15 15:02:34
100  :  AutoModerator
2015-09-15 20:03:29
100  :  AutoModerator
2015-09-15 23:22:09
100  :  AutoModerator
2015-09-16 08:32:34
100  :  AutoModerator
2015-09-16 15:56:10
100  :  AutoModerator
2015-09-16 21:02:28
100  :  AutoModerator
2015-09-17 00:00:14
100  :  AutoModerator
2015-09-17 07:03:25
100  :  AutoModerator
2015-09-17 16:02:29
100  :  AutoModerator
2015-09-17 21:01:26
100  :  AutoModerator
2015-09-18 01:02:12
100  :  AutoModerator
2015-09-18 09:04:21
100  :  AutoModerator
2015-09-18 16:00:18
100  :  AutoModerator
2015-09-18 19:06:05
100  :  AutoModerator
2015-09-18 22:04:25
100  :  AutoModerator
2015-09-19 00:08:03
100  :  AutoModerator
2015-09-19 07:03:44
100  :  AutoModerator
2015-09-19 14:56:22
100  :  AutoModerator
2015-09-19 20:02:22
100  :  AutoModerator
2015-09-20 0

3  :  Bearsaregood
2016-12-08 11:50:29
1  :  Beautiful_dreamer123
2019-02-23 01:32:37
1  :  Becky-and-Momo
2019-02-11 04:48:20
39  :  BeezyWeezyWoo
2019-06-06 01:39:22
19  :  Bekind123
2018-09-11 14:40:17
4  :  Benfoo
2019-07-23 02:57:17
1  :  Benji_Boy13
2017-09-29 08:57:05
2  :  Benz1311
2018-08-03 04:38:41
31  :  BigFig32
2018-09-26 06:01:58
44  :  BigLittleSloth
2019-04-19 18:57:48
3  :  Big_Tuna13
2018-05-16 14:47:42
49  :  BiggerInTheInside
2018-01-01 21:40:06
1  :  Billy-Wilder-97
2017-05-07 09:00:43
1  :  BisRack
2016-09-16 16:23:02
10  :  Bleeper2000
2020-01-12 23:55:00
95  :  Bloodborne505
2020-02-24 06:32:52
54  :  Bludstained
2015-05-15 12:24:20
32  :  Bluelightt
2017-12-22 05:50:23
2  :  Bluetalion
2019-10-10 21:17:11
3  :  BlxckCat
2019-05-21 12:51:20
2  :  Bnerna
2019-09-08 13:45:56
100  :  BodhisattvaJones
2014-02-02 05:08:11
68  :  BodhisattvaJones
2015-01-05 08:04:04
3  :  Boggie135
2019-07-05 22:24:53
57  :  BonglordFourTwenny
2018-01-03 23:38:15
43  :  BoostedBenji


100  :  Ethas
2017-06-09 14:12:41
76  :  Ethas
2017-11-19 02:06:26
34  :  Existence1290
2019-04-24 23:02:18
13  :  ExplosiveTumbleweed
2018-12-14 18:12:14
4  :  Extreme_apathy_late
2019-07-26 10:06:42
100  :  EyeOfSol
2016-01-04 02:32:05
5  :  EyeOfSol
2016-01-14 20:39:31
100  :  F00dbAby
2017-08-12 13:40:06
100  :  F00dbAby
2017-09-17 12:40:10
100  :  F00dbAby
2017-10-15 19:10:54
85  :  F00dbAby
2017-11-20 21:35:56
6  :  FLlove89
2018-05-03 12:50:13
36  :  FRP7
2019-06-21 20:53:35
2  :  Facelessmonster
2017-10-24 19:04:30
51  :  Failure1234567
2020-01-04 16:41:52
6  :  FantasticNature1
2020-05-22 02:22:20
5  :  Fashionistafor20
2018-05-22 12:30:59
62  :  FatButLittle
2018-11-04 12:42:04
5  :  Faustty
2018-08-17 21:42:37
18  :  FeatheredDrake
2017-08-17 21:09:51
18  :  FeedMeMau5
2017-05-08 10:08:44
1  :  Feelofme
2019-01-10 12:05:10
5  :  Fick_Thingers
2017-10-07 18:35:31
1  :  FidgetSp1nner
2019-11-21 17:56:36
100  :  FieldOfViews
2019-07-15 04:23:52
13  :  FieldOfViews
2019-07-18 03

100  :  KF2015
2017-04-14 03:08:20
100  :  KF2015
2017-06-15 14:30:08
46  :  KF2015
2017-08-16 10:14:27
20  :  KHCross
2018-10-06 12:57:14
13  :  KaleMunoz
2017-04-17 12:56:15
1  :  KasperAura
2016-08-16 00:45:05
68  :  Kass1207
2013-12-08 13:12:45
54  :  Kaylafish
2015-08-03 15:10:06
39  :  KeepYourClawsOut
2017-09-22 04:55:36
7  :  Keith_H23
2018-08-21 02:11:43
6  :  KekyPowa
2017-08-01 19:54:54
3  :  Kennytheanxietykid
2017-12-01 06:38:33
65  :  Kennyv777
2017-03-13 15:28:25
1  :  Kerrypug
2016-12-24 02:24:56
100  :  KetoTeacher12
2018-08-04 03:35:06
100  :  KetoTeacher12
2018-12-08 10:09:02
100  :  KetoTeacher12
2019-04-02 11:39:26
100  :  KetoTeacher12
2019-09-11 11:26:15
64  :  KetoTeacher12
2019-11-04 09:15:28
6  :  KevinTheKoop
2017-06-17 08:20:17
12  :  KhaosHiDef
2016-12-18 03:05:44
14  :  KidQuesadilla
2020-01-15 11:08:19
15  :  Kimoly
2018-01-11 04:34:33
2  :  KindergartenRolex
2018-07-12 00:37:57
2  :  KingfisherClaws
2019-03-17 21:54:26
42  :  Kissessuckmytoes
2017-11-23 

100  :  Nobody_Likes_Shy_Guy
2020-01-12 02:21:21
100  :  Nobody_Likes_Shy_Guy
2020-05-28 06:22:38
38  :  Nobody_Likes_Shy_Guy
2020-06-20 03:11:22
3  :  Normal-Anxious
2020-02-26 15:36:18
4  :  NorthNorne
2019-01-20 10:16:34
1  :  NorthernBCGuy
2018-04-08 12:52:49
43  :  Northern_Chiliad
2015-12-28 19:56:08
5  :  NotInVogue
2017-08-06 05:20:25
100  :  NotUrAvrageFish
2016-11-09 09:28:33
100  :  NotUrAvrageFish
2017-01-26 03:12:13
79  :  NotUrAvrageFish
2017-04-08 02:41:58
4  :  NovaFalconGamer
2017-10-28 10:42:30
10  :  Novemberx123
2019-10-05 03:00:50
48  :  Number175OnEarlsList
2017-11-30 11:52:39
9  :  NuttingWithTheForce
2018-09-25 19:25:40
5  :  NyaRice
2015-02-26 10:50:07
1  :  Nykitty123
2019-06-25 22:34:34
100  :  NythilMahariel
2018-01-10 06:10:59
100  :  NythilMahariel
2018-08-03 13:18:14
100  :  NythilMahariel
2019-01-30 19:11:04
3  :  NythilMahariel
2019-02-13 18:11:53
2  :  OCDawayaway
2017-06-17 14:46:53
100  :  ODMAN03
2018-04-14 02:20:20
100  :  ODMAN03
2018-08-09 07:49:

10  :  SmittyManJensen_
2018-10-01 03:35:45
16  :  SmollIndividual
2020-04-01 11:02:08
79  :  Snitchwithabitch
2019-08-07 00:35:50
13  :  Soap-Taste-Ok
2017-12-28 07:00:03
74  :  Some1inreallife
2020-04-28 09:00:35
2  :  SomeGuy1021
2019-06-06 13:56:42
1  :  SomeGuyMe
2017-05-26 03:41:38
16  :  Somerandomdickhead
2018-03-19 13:30:14
27  :  Songgeek
2018-08-29 11:43:05
27  :  Sophharvs
2019-10-22 05:29:52
18  :  SorryImBadWithNames
2019-10-19 14:15:09
3  :  Spartan8080
2019-06-30 06:12:19
2  :  SpicyMemeLorde
2018-12-14 12:54:51
14  :  Spicyhotdogfarts
2017-03-22 13:56:35
100  :  SpongeTheOC
2017-02-24 22:21:13
3  :  SpongeTheOC
2017-02-27 09:37:46
59  :  Spongemage
2017-12-04 07:19:03
33  :  SquareThings
2018-07-27 11:03:50
100  :  Squishy_Pixelz
2017-01-03 22:59:32
100  :  Squishy_Pixelz
2017-06-13 08:25:15
100  :  Squishy_Pixelz
2017-12-03 09:43:36
100  :  Squishy_Pixelz
2018-04-03 08:58:39
65  :  Squishy_Pixelz
2018-06-13 09:34:40
3  :  Stackz18
2019-08-09 16:00:58
13  :  StalinsThi

4  :  WeeklyAddition
2018-09-12 01:07:34
1  :  Weenars
2016-11-11 07:04:35
3  :  Weirdest_Llama_Ever
2018-09-11 18:08:14
5  :  Weirdoz96
2018-01-16 02:52:16
100  :  Wh00pity_sc00p
2019-06-24 02:08:30
37  :  Wh00pity_sc00p
2019-08-26 12:50:56
7  :  WhereAKidCanBeAKid
2018-07-08 15:50:42
2  :  WhichMonkeyIsBest
2017-11-28 04:44:48
100  :  Wholesome_Doinks
2018-12-26 21:23:46
40  :  Wholesome_Doinks
2019-03-02 06:50:31
2  :  WindUpTomato
2016-02-18 13:19:16
71  :  WindowsKidd
2018-03-17 17:52:52
1  :  WinduFin
2019-07-13 23:46:53
5  :  Winnie70823
2020-05-10 10:23:36
2  :  WinterWitch89
2018-06-27 23:50:08
4  :  Wisso94
2018-09-07 04:57:48
100  :  Woddie_321
2017-10-15 00:16:00
100  :  Woddie_321
2018-03-03 20:22:22
100  :  Woddie_321
2018-04-30 22:02:43
100  :  Woddie_321
2018-05-25 03:43:55
100  :  Woddie_321
2018-06-22 17:29:56
99  :  Woddie_321
2018-09-18 21:42:25
100  :  Woddie_321
2018-11-19 00:08:30
100  :  Woddie_321
2019-02-23 18:14:39
36  :  Woddie_321
2019-05-28 04:22:45
2  :  

66  :  bawan15
2018-11-05 18:49:41
100  :  bcxx143
2016-06-05 02:55:21
65  :  bcxx143
2016-09-02 08:45:42
100  :  bdstwin
2018-01-06 03:13:36
100  :  bdstwin
2018-05-07 06:22:42
100  :  bdstwin
2018-06-25 09:34:53
100  :  bdstwin
2018-09-19 07:22:40
100  :  bdstwin
2019-01-18 11:09:51
81  :  bdstwin
2019-08-31 08:34:22
6  :  beaniee67
2017-10-13 09:12:40
9  :  beanxheerioz
2019-11-02 12:53:27
8  :  beariesad
2019-05-26 09:28:17
57  :  beautifulexistence
2015-06-21 17:56:11
2  :  beautifulmessssss
2017-08-04 11:05:31
27  :  beautynsc
2014-12-30 21:09:09
3  :  beencouraged
2018-03-20 13:43:01
2  :  beinn_fiadhaich
2018-01-07 06:00:03
1  :  believablykind
2019-09-03 22:42:58
14  :  belladoll1021
2019-04-16 07:56:39
4  :  bendieben
2019-02-15 13:21:32
9  :  bentleyherman
2019-09-26 12:47:34
2  :  bestdayever31
2018-09-13 03:27:15
1  :  bibisleep
2019-11-04 16:55:52
1  :  biggerperspective
2019-04-21 08:41:29
4  :  bigjuiceyoman
2017-03-16 04:43:52
100  :  bigstarlivealittle
2019-04-08 10:0

100  :  elreydelosgueys
2019-10-25 00:52:17
40  :  elreydelosgueys
2019-11-09 04:24:56
52  :  emerald188
2019-09-29 08:03:26
40  :  emza555
2019-01-13 11:33:38
24  :  enricowow
2019-06-09 17:21:33
41  :  enzanurOz
2018-07-02 07:04:30
4  :  eoininin
2016-03-10 10:24:20
1  :  eonblue7
2013-11-26 05:25:32
3  :  epilithic
2018-10-24 12:01:04
100  :  erikag02
2019-02-04 07:49:31
100  :  erikag02
2019-04-06 13:17:17
100  :  erikag02
2019-06-13 12:20:20
100  :  erikag02
2019-09-08 06:49:59
21  :  erikag02
2019-09-27 07:20:51
10  :  espressodaydream
2018-03-26 23:56:16
['Sir_Biscuit', 'eternalplatoon']
68  :  eugenedhartke
2020-01-05 11:01:13
2  :  eunbis
2019-09-09 19:23:27
14  :  eva1588
2018-08-23 00:55:40
26  :  evude8
2018-06-11 06:35:05
9  :  evvaanxox
2019-01-16 05:59:01
4  :  explodedgiraffe
2017-08-31 10:58:28
35  :  extasis_T
2017-12-05 15:13:46
4  :  faisalrishu
2019-04-12 15:17:22
49  :  fantahsmic
2019-07-31 10:30:43
21  :  fatboyslimzzz
2018-11-25 08:31:33
51  :  felipe5083
2018-

36  :  jokamp
2018-01-11 14:10:00
3  :  jonesbrooklyn773
2019-07-25 11:30:41
100  :  joyous_occlusion
2016-07-08 00:09:24
23  :  joyous_occlusion
2017-03-20 12:45:21
19  :  jrt364
2016-01-17 07:59:38
1  :  jssbklap
2020-02-21 00:02:42
6  :  jthomas2234
2018-11-20 12:00:10
12  :  juiceepeach
2019-09-03 05:21:02
79  :  juicycouturexoxo2006
2017-06-25 08:44:23
40  :  justausername92
2019-07-25 15:22:13
1  :  justdrapin
2018-08-07 14:38:54
3  :  justforaseason
2020-02-20 12:30:50
17  :  juststupid8970
2019-03-03 10:12:58
16  :  k0alabearsleeping
2019-04-17 11:16:42
1  :  k3llyl0rraine89
2017-10-03 04:25:00
5  :  kaidorke
2018-08-16 13:58:12
100  :  kaizokukaiju
2019-03-17 09:00:00
28  :  kaizokukaiju
2019-04-29 11:59:24
1  :  kajjaznam1
2017-01-23 01:03:51
3  :  kalakakes
2019-10-25 18:29:56
16  :  kaos424
2015-09-03 12:20:29
52  :  karnerblu
2015-12-25 06:20:33
17  :  katiecat007
2019-10-01 02:37:36
10  :  katiecski
2017-11-20 19:39:20
12  :  katja_x
2019-05-07 15:32:50
1  :  kaxxcxn
2016

100  :  mustardyellow123
2018-08-24 07:30:14
40  :  mustardyellow123
2019-04-09 14:02:33
29  :  mustydusterman
2018-01-04 21:59:49
100  :  mvxn
2018-04-07 06:30:47
100  :  mvxn
2018-05-17 15:59:14
100  :  mvxn
2018-06-18 00:20:26
100  :  mvxn
2018-07-26 14:40:37
100  :  mvxn
2018-09-19 13:25:00
54  :  mvxn
2018-10-08 14:50:08
5  :  mxcuxs
2018-08-29 07:00:33
25  :  myconfessionacc
2015-06-15 03:26:19
1  :  mykingdomforsleep
2016-08-10 15:25:11
11  :  mymentalhealthisbad
2020-05-28 14:22:26
4  :  mysacrifice13
2019-03-04 16:54:50
4  :  n3ssypoo
2018-02-05 06:26:45
67  :  n8toxic21
2017-08-14 10:32:39
7  :  naivecarrot420
2018-08-08 21:17:41
4  :  nannygal3
2017-12-31 07:15:59
8  :  nanotech23
2016-10-11 07:35:54
4  :  natemp21
2019-06-19 04:40:04
35  :  naylsonsb
2018-11-15 21:58:59
10  :  nebulochaotic_
2019-12-06 07:23:25
43  :  needleworkreverie
2018-08-13 04:29:02
78  :  neosharks
2019-03-14 11:21:13
100  :  nerdforeverrrrrr
2019-11-11 01:37:30
62  :  nerdforeverrrrrr
2020-01-15 00:

45  :  saultarus
2020-01-16 11:08:47
8  :  scarpmclovin
2018-05-08 08:18:55
4  :  sceerdycat
2019-06-16 00:28:30
10  :  sco8881
2020-04-07 18:27:16
40  :  scurtel
2019-02-16 06:27:50
3  :  seahunter1815
2018-07-08 23:42:24
100  :  sean7755
2013-11-13 11:21:21
43  :  sean7755
2013-11-22 12:41:21
6  :  seekingstars
2018-01-09 09:20:17
1  :  seniilisiili
2020-02-24 19:36:07
15  :  serendipity1996
2016-06-08 07:14:01
6  :  seven2812
2015-09-15 10:59:39
7  :  sflo97
2018-12-05 12:49:56
1  :  shadowgun190
2017-01-23 15:09:16
19  :  shadowofassassin
2018-11-02 08:17:21
3  :  shakespearestark
2017-04-05 12:53:25
1  :  shakeyaaboooty
2017-05-03 03:46:20
32  :  shakingunder
2016-03-20 14:35:36
6  :  sheeniebeanie1
2019-04-23 08:36:22
1  :  shewithdrewneatly
2016-01-31 15:45:03
52  :  shockrush
2017-03-17 06:57:21
5  :  shoshana20
2017-08-12 12:50:59
3  :  shru4i97
2018-03-09 13:50:06
5  :  shutupmyhead
2019-05-27 06:44:19
7  :  shybi_librarian
2016-12-13 12:59:21
9  :  sighofwinter
2019-11-04 17

100  :  unofficialmoderator
2018-06-07 20:30:32
100  :  unofficialmoderator
2018-07-02 07:13:25
100  :  unofficialmoderator
2018-07-28 08:27:30
100  :  unofficialmoderator
2018-08-09 19:13:25
94  :  unofficialmoderator
2018-09-04 04:32:11
100  :  unofficialmoderator
2018-09-26 06:49:33
100  :  unofficialmoderator
2018-10-26 05:47:10
73  :  unofficialmoderator
2018-11-23 08:31:18
1  :  unseentides
2015-11-02 07:56:53
26  :  user_7061
2019-12-18 17:13:32
9  :  username-taken1
2015-11-05 22:51:05
16  :  usernametakenkappa
2014-11-05 18:35:06
8  :  uszn_in
2018-09-30 05:40:30
5  :  vaasan
2019-11-17 19:59:22
4  :  valentinagd01
2020-04-09 14:03:29
100  :  valueyoghurt
2018-12-05 05:44:42
12  :  valueyoghurt
2019-01-15 08:36:54
3  :  vanessawinters
2018-03-29 13:19:44
9  :  veganburritokiller
2017-10-07 11:37:09
2  :  veggietown4
2020-04-17 14:52:09
1  :  veraloe
2018-04-24 09:05:10
6  :  veryberryblue
2018-07-25 06:08:38
49  :  victorsledge07
2016-10-12 02:48:56
17  :  vietusernam
2016-05-

In [33]:
deleted_author = a[a['author'] == '[deleted]']
a = a.drop(deleted_author.index)
a = a.drop(a[a['selftext'] == '[deleted]'].index)
a = a.drop(a[a['selftext'] == '[removed]'].index)
content_null = a[a['selftext'].isnull()].index
a= a.drop(content_null)

In [34]:
a.to_csv("user_data\\schizophrenia_author_before.csv", index= False)

In [35]:
a

subreddit                                              title  \
0            opiates                      Opiates just to numb feelings   
1            AskDocs                    Why am I feel sick, on and off?   
2      schizophrenia                                               Why?   
4             oculus  How to play GTA5 in Oculus CV1 with head track...   
5      schizophrenia  Can anyone advise me in choosing my second dru...   
...              ...                                                ...   
15285        23andme  Hello all please look at some of my results fo...   
15286        23andme  What site can i upload my genomind data? I wan...   
15289         Queens  Has anyone done stem cell to cure there depres...   
15292            OCD  What is better advice than just saying exercis...   
15293  schizophrenia  What is better advice than just saying exercis...   

              author                                           selftext  \
0           -Renton-  I don't even know what I'm doing anymore, opia...   
1           -Renton-  Lately I've been having really on and off naus...   
2           -Renton-  So why are people wanting to kidnap me and my ...   
4      0-01percenter  I'd like to know if there's such possibility, ...   
5      0-01percenter  I'm going to my first visit tomorrow and wonde...   
...              ...                                                ...   
15285       zumifi96  hello so according to my genomind report on th...   
15286       zumifi96  What site can i upload my genomind data? I wan...   
15289       zumifi96  Has anyone done stem cell to cure there depres...   
15292       zumifi96  What is better advice than just saying exercis...   
15293       zumifi96  What is better advice than just saying exercis...   

                      date  subredddit_check  
0      2014-08-10 05:58:31                 0  
1      2014-08-16 02:07:31                 0  
2      2014-08-18 07:14:15                 1  
4      2016-10-10 21:59:21                 0  
5      2016-10-19 02:28:07                 1  
...                    ...               ...  
15285  2018-02-15 12:46:37                 0  
15286  2018-02-16 02:53:59                 0  
15289  2018-02-17 09:17:50                 0  
15292  2018-02-18 07:54:01                 0  
15293  2018-02-18 08:03:58                 1  

[9148 rows x 6 columns]

## 4) author별 게시글 수집

정신질환에 글을 게시한 시점으로 다른 하위 레딧 

In [3]:
import json
def getPushshiftData(after, before, author):
    url = 'https://api.pushshift.io/reddit/search/submission/?size=1000&after='+str(after)+'&before='+str(before)+'&author='+str(author)
    r = requests.get(url)
    data = json.loads(r.text)
    return data['data']

In [4]:
def collectSubData(subm): # 글전체 갖고오기 ->이거 하는게 제일 낫긴 함
    subData = list() #list to store data points
    title = subm['title']
    try:
        selftext = subm['selftext']
    except:
        selftext = 'none'
    finally:
        subreddit = subm['subreddit']
        author = subm['author']
        sub_id = subm['id']
        if(subreddit == "anxiety"): ##################병명 변경#############################
            subredddit_check = 1
        else:
            subredddit_check = 0
        created = datetime.datetime.fromtimestamp(subm['created_utc'])
        subData.append((subreddit,title,author,selftext,created, subredddit_check))
        subStats[sub_id] = subData

In [5]:
# 사용자별 최초작성일 저장 csv 읽어오기
author_df = pd.read_csv("anxiety\\author_firstdate_anxiety.csv")

In [ ]:
start = time.time()
error_author = [] 
#가끔 에러가 나는 케이스도 있어서 일단 별도로 저장해두는 리스트입니당
subStats = {}
for i in range(len(author_df)):
    author = author_df['author'].iloc[i]
    try:
        after = int(convert_unixtime(author_df['first_date'].iloc[i]))
        before = 1600831376
        # 파일 수집
        subCount = 0
        data = getPushshiftData(after, before, author)
        # Will run until all posts have been gathered 
        # from the 'after' date up until before date
        while len(data) > 0:
            for submission in data:
                collectSubData(submission)
                subCount+=1
            # Calls getPushshiftData() with the created date of the last submission
            print(len(data), " : ", author)
            print(str(datetime.datetime.fromtimestamp(data[-1]['created_utc'])))
            after = data[-1]['created_utc']
            while True:
                try:
                    data = getPushshiftData(after, before, author)
                    break
                except:
                    print('에러 발생')
                    continue
    except:
        error_author.append(author)
        print(error_author)

                    
# 파일 저장
f = open("anxiety\\anxiety_author_after.csv",'a',encoding='utf-8-sig') ##################병명 변경#############################
wr = csv.writer(f)
wr.writerow(['subreddit', 'title', 'author', 'selftext', "date", 'subreddit_check'])
for i in list(subStats.values()):
    wr.writerow(i[0])
f.close()
print(author, " 파일 저장 성공!")
print('걸린시간:'+ str(time.time()-start))

93  :  -Skelan-
2020-09-22 07:23:24
73  :  -bannedaccount-
2020-07-14 10:41:10
100  :  1000ancestors
2017-07-05 13:16:24
100  :  1000ancestors
2019-03-11 13:44:32
100  :  1000ancestors
2020-02-10 10:00:14
52  :  1000ancestors
2020-09-16 13:24:53
70  :  104degrees_cold
2020-09-04 05:46:11
100  :  1234swkisgar56
2018-11-19 19:01:02
100  :  1234swkisgar56
2020-03-30 15:49:29
65  :  1234swkisgar56
2020-09-22 07:37:34
40  :  123anxiety
2017-09-01 00:08:20
100  :  123grimey
2018-09-30 00:34:09
100  :  123grimey
2020-05-06 16:27:02
13  :  123grimey
2020-09-09 23:02:47
66  :  123likeabirdimfree
2020-08-20 10:57:02
100  :  1398_Days
2017-04-12 13:39:04
100  :  1398_Days
2018-06-18 08:24:30
99  :  1398_Days
2018-12-03 07:56:15
100  :  1398_Days
2019-05-20 06:17:44
100  :  1398_Days
2019-11-03 04:10:09
100  :  1398_Days
2020-03-28 04:23:28
100  :  1398_Days
2020-07-23 05:27:39
47  :  1398_Days
2020-09-23 05:34:06
100  :  13yearoldidiot
2019-06-29 04:35:15
100  :  13yearoldidiot
2019-07-30 01:26:5

100  :  ArtisticAnxiety
2018-05-15 10:57:55
100  :  ArtisticAnxiety
2019-06-03 02:14:22
33  :  ArtisticAnxiety
2020-09-13 05:16:58
100  :  Artistic_Inflation
2019-06-06 10:26:52
100  :  Artistic_Inflation
2019-07-10 13:25:41
100  :  Artistic_Inflation
2020-03-03 00:12:41
14  :  Artistic_Inflation
2020-04-13 06:34:22
100  :  Asgarnian_Seed
2019-07-22 13:28:58
94  :  Asgarnian_Seed
2020-01-27 13:43:48
41  :  AshNJewells
2019-12-29 16:00:03
100  :  Ashh_The_CyborgWitch
2020-01-27 04:32:44
100  :  Ashh_The_CyborgWitch
2020-05-13 13:31:49
41  :  Ashh_The_CyborgWitch
2020-06-26 03:40:57
100  :  AshleyJames_
2020-07-27 22:53:53
38  :  AshleyJames_
2020-09-21 01:54:36
23  :  Assilem113
2018-12-15 11:57:30
100  :  Assuagehempcbd
2020-03-03 09:56:51
3  :  Assuagehempcbd
2020-04-10 00:39:38
100  :  Aturdinatuba
2019-12-31 07:00:01
73  :  Aturdinatuba
2020-02-26 14:18:59
100  :  Auri15
2018-08-05 03:55:23
41  :  Auri15
2019-12-02 02:42:39
52  :  Aussiegal33
2018-12-29 09:50:32
100  :  AutoModerato

100  :  AutoModerator
2015-11-30 01:04:20
100  :  AutoModerator
2015-11-30 08:02:30
100  :  AutoModerator
2015-11-30 13:00:22
100  :  AutoModerator
2015-11-30 16:05:27
100  :  AutoModerator
2015-11-30 19:10:34
100  :  AutoModerator
2015-11-30 21:32:11
100  :  AutoModerator
2015-12-01 00:05:02
100  :  AutoModerator
2015-12-01 05:03:15
100  :  AutoModerator
2015-12-01 12:03:27
100  :  AutoModerator
2015-12-01 17:02:42
100  :  AutoModerator
2015-12-01 20:04:00
100  :  AutoModerator
2015-12-01 23:02:47
100  :  AutoModerator
2015-12-02 03:02:52
100  :  AutoModerator
2015-12-02 11:03:17
100  :  AutoModerator
2015-12-02 17:02:34
100  :  AutoModerator
2015-12-02 21:03:43
100  :  AutoModerator
2015-12-03 00:03:53
100  :  AutoModerator
2015-12-03 06:40:22
100  :  AutoModerator
2015-12-03 13:58:12
100  :  AutoModerator
2015-12-03 19:03:26
100  :  AutoModerator
2015-12-03 22:05:27
100  :  AutoModerator
2015-12-04 03:03:01
100  :  AutoModerator
2015-12-04 11:02:18
100  :  AutoModerator
2015-12-04 1

100  :  AutoModerator
2016-01-08 15:02:22
100  :  AutoModerator
2016-01-08 18:03:43
100  :  AutoModerator
2016-01-08 21:04:21
100  :  AutoModerator
2016-01-08 23:04:16
100  :  AutoModerator
2016-01-09 02:02:45
100  :  AutoModerator
2016-01-09 09:02:40
100  :  AutoModerator
2016-01-09 15:02:06
100  :  AutoModerator
2016-01-09 19:32:17
100  :  AutoModerator
2016-01-09 23:00:22
100  :  AutoModerator
2016-01-10 04:16:21
100  :  AutoModerator
2016-01-10 12:02:10
100  :  AutoModerator
2016-01-10 17:02:15
100  :  AutoModerator
2016-01-10 21:02:46
100  :  AutoModerator
2016-01-11 00:02:38
100  :  AutoModerator
2016-01-11 04:32:44
100  :  AutoModerator
2016-01-11 09:08:27
100  :  AutoModerator
2016-01-11 15:00:08
100  :  AutoModerator
2016-01-11 17:05:06
100  :  AutoModerator
2016-01-11 20:03:53
100  :  AutoModerator
2016-01-11 22:03:16
100  :  AutoModerator
2016-01-12 00:05:03
100  :  AutoModerator
2016-01-12 05:16:18
100  :  AutoModerator
2016-01-12 12:32:09
100  :  AutoModerator
2016-01-12 1

100  :  AutoModerator
2016-02-12 09:03:20
100  :  AutoModerator
2016-02-12 14:00:08
100  :  AutoModerator
2016-02-12 17:03:06
100  :  AutoModerator
2016-02-12 20:04:28
100  :  AutoModerator
2016-02-12 22:04:09
100  :  AutoModerator
2016-02-13 00:04:28
100  :  AutoModerator
2016-02-13 03:30:24
100  :  AutoModerator
2016-02-13 10:02:22
100  :  AutoModerator
2016-02-13 15:46:17
100  :  AutoModerator
2016-02-13 20:02:13
100  :  AutoModerator
2016-02-13 22:42:21
100  :  AutoModerator
2016-02-14 03:42:20
100  :  AutoModerator
2016-02-14 10:02:39
100  :  AutoModerator
2016-02-14 16:02:04
100  :  AutoModerator
2016-02-14 20:02:47
100  :  AutoModerator
2016-02-14 23:00:36
100  :  AutoModerator
2016-02-15 02:03:04
100  :  AutoModerator
2016-02-15 09:00:22
100  :  AutoModerator
2016-02-15 12:02:08
100  :  AutoModerator
2016-02-15 16:02:44
100  :  AutoModerator
2016-02-15 18:42:20
100  :  AutoModerator
2016-02-15 21:05:10
100  :  AutoModerator
2016-02-15 23:00:55
100  :  AutoModerator
2016-02-16 0

100  :  AutoModerator
2016-03-16 21:02:11
100  :  AutoModerator
2016-03-16 23:02:28
100  :  AutoModerator
2016-03-17 02:02:23
100  :  AutoModerator
2016-03-17 09:02:03
100  :  AutoModerator
2016-03-17 15:02:34
100  :  AutoModerator
2016-03-17 19:52:13
100  :  AutoModerator
2016-03-17 22:16:16
100  :  AutoModerator
2016-03-18 02:02:06
100  :  AutoModerator
2016-03-18 09:02:14
100  :  AutoModerator
2016-03-18 14:02:30
100  :  AutoModerator
2016-03-18 17:32:10
100  :  AutoModerator
2016-03-18 21:00:23
100  :  AutoModerator
2016-03-18 22:02:38
100  :  AutoModerator
2016-03-19 00:02:49
100  :  AutoModerator
2016-03-19 05:02:20
100  :  AutoModerator
2016-03-19 11:02:25
100  :  AutoModerator
2016-03-19 16:44:08
100  :  AutoModerator
2016-03-19 20:16:15
100  :  AutoModerator
2016-03-19 23:02:22
100  :  AutoModerator
2016-03-20 05:02:13
100  :  AutoModerator
2016-03-20 12:58:19
100  :  AutoModerator
2016-03-20 17:02:13
100  :  AutoModerator
2016-03-20 21:02:09
100  :  AutoModerator
2016-03-20 2

100  :  AutoModerator
2016-04-18 03:46:12
100  :  AutoModerator
2016-04-18 09:06:03
100  :  AutoModerator
2016-04-18 13:03:57
100  :  AutoModerator
2016-04-18 16:04:33
100  :  AutoModerator
2016-04-18 18:06:12
100  :  AutoModerator
2016-04-18 20:46:18
100  :  AutoModerator
2016-04-18 22:05:00
100  :  AutoModerator
2016-04-19 00:02:48
100  :  AutoModerator
2016-04-19 02:56:12
100  :  AutoModerator
2016-04-19 09:03:07
100  :  AutoModerator
2016-04-19 14:04:11
100  :  AutoModerator
2016-04-19 18:04:10
100  :  AutoModerator
2016-04-19 21:05:04
100  :  AutoModerator
2016-04-20 00:03:22
100  :  AutoModerator
2016-04-20 06:02:57
100  :  AutoModerator
2016-04-20 12:58:17
100  :  AutoModerator
2016-04-20 16:06:18
100  :  AutoModerator
2016-04-20 20:03:25
100  :  AutoModerator
2016-04-20 22:05:23
100  :  AutoModerator
2016-04-21 00:06:34
100  :  AutoModerator
2016-04-21 06:02:20
100  :  AutoModerator
2016-04-21 12:02:06
100  :  AutoModerator
2016-04-21 17:03:19
100  :  AutoModerator
2016-04-21 2

100  :  AutoModerator
2016-05-17 19:56:30
100  :  AutoModerator
2016-05-17 22:00:24
100  :  AutoModerator
2016-05-18 01:02:21
100  :  AutoModerator
2016-05-18 06:02:44
100  :  AutoModerator
2016-05-18 12:02:31
100  :  AutoModerator
2016-05-18 16:03:00
100  :  AutoModerator
2016-05-18 19:03:41
100  :  AutoModerator
2016-05-18 21:07:18
100  :  AutoModerator
2016-05-18 23:06:39
100  :  AutoModerator
2016-05-19 02:05:26
100  :  AutoModerator
2016-05-19 08:56:22
100  :  AutoModerator
2016-05-19 13:26:18
100  :  AutoModerator
2016-05-19 17:16:19
100  :  AutoModerator
2016-05-19 21:02:41
100  :  AutoModerator
2016-05-19 23:05:03
100  :  AutoModerator
2016-05-20 02:05:09
100  :  AutoModerator
2016-05-20 09:02:23
100  :  AutoModerator
2016-05-20 13:03:38
100  :  AutoModerator
2016-05-20 16:05:39
100  :  AutoModerator
2016-05-20 19:03:43
100  :  AutoModerator
2016-05-20 21:05:26
100  :  AutoModerator
2016-05-20 22:07:50
100  :  AutoModerator
2016-05-21 00:05:16
100  :  AutoModerator
2016-05-21 0

100  :  AutoModerator
2016-06-14 23:04:38
100  :  AutoModerator
2016-06-15 03:32:20
100  :  AutoModerator
2016-06-15 09:16:18
100  :  AutoModerator
2016-06-15 15:03:02
100  :  AutoModerator
2016-06-15 18:02:05
100  :  AutoModerator
2016-06-15 21:02:40
100  :  AutoModerator
2016-06-15 23:02:35
100  :  AutoModerator
2016-06-16 01:04:21
100  :  AutoModerator
2016-06-16 07:02:49
100  :  AutoModerator
2016-06-16 13:03:19
100  :  AutoModerator
2016-06-16 17:04:26
100  :  AutoModerator
2016-06-16 20:04:56
100  :  AutoModerator
2016-06-16 23:02:20
100  :  AutoModerator
2016-06-17 02:02:11
100  :  AutoModerator
2016-06-17 09:02:11
100  :  AutoModerator
2016-06-17 13:03:57
100  :  AutoModerator
2016-06-17 16:05:04
100  :  AutoModerator
2016-06-17 18:48:10
100  :  AutoModerator
2016-06-17 21:02:33
100  :  AutoModerator
2016-06-17 22:06:28
100  :  AutoModerator
2016-06-18 00:04:30
100  :  AutoModerator
2016-06-18 02:56:35
100  :  AutoModerator
2016-06-18 09:03:42
100  :  AutoModerator
2016-06-18 1

100  :  AutoModerator
2016-07-13 16:05:44
100  :  AutoModerator
2016-07-13 19:04:42
100  :  AutoModerator
2016-07-13 21:32:24
100  :  AutoModerator
2016-07-14 00:02:26
100  :  AutoModerator
2016-07-14 02:03:54
100  :  AutoModerator
2016-07-14 08:52:20
100  :  AutoModerator
2016-07-14 14:00:24
100  :  AutoModerator
2016-07-14 17:05:35
100  :  AutoModerator
2016-07-14 20:05:13
100  :  AutoModerator
2016-07-14 23:02:31
100  :  AutoModerator
2016-07-15 02:02:44
100  :  AutoModerator
2016-07-15 09:02:05
100  :  AutoModerator
2016-07-15 13:02:23
100  :  AutoModerator
2016-07-15 16:04:36
100  :  AutoModerator
2016-07-15 18:04:58
100  :  AutoModerator
2016-07-15 20:12:17
100  :  AutoModerator
2016-07-15 22:04:57
100  :  AutoModerator
2016-07-15 23:46:25
100  :  AutoModerator
2016-07-16 02:02:25
100  :  AutoModerator
2016-07-16 07:03:58
100  :  AutoModerator
2016-07-16 12:56:27
100  :  AutoModerator
2016-07-16 16:44:07
100  :  AutoModerator
2016-07-16 19:56:25
100  :  AutoModerator
2016-07-16 2

100  :  AutoModerator
2016-08-08 18:16:18
100  :  AutoModerator
2016-08-08 20:07:20
100  :  AutoModerator
2016-08-08 21:32:17
100  :  AutoModerator
2016-08-08 23:04:26
100  :  AutoModerator
2016-08-09 01:02:17
100  :  AutoModerator
2016-08-09 03:22:18
100  :  AutoModerator
2016-08-09 08:28:12
100  :  AutoModerator
2016-08-09 12:50:16
100  :  AutoModerator
2016-08-09 16:04:00
100  :  AutoModerator
2016-08-09 19:03:57
100  :  AutoModerator
2016-08-09 21:06:14
100  :  AutoModerator
2016-08-09 23:06:24
100  :  AutoModerator
2016-08-10 02:56:30
100  :  AutoModerator
2016-08-10 08:03:36
100  :  AutoModerator
2016-08-10 12:32:14
100  :  AutoModerator
2016-08-10 15:56:14
100  :  AutoModerator
2016-08-10 18:03:35
100  :  AutoModerator
2016-08-10 21:03:05
100  :  AutoModerator
2016-08-10 23:02:52
100  :  AutoModerator
2016-08-11 00:52:29
100  :  AutoModerator
2016-08-11 04:20:19
100  :  AutoModerator
2016-08-11 09:04:18
100  :  AutoModerator
2016-08-11 14:00:26
100  :  AutoModerator
2016-08-11 1

100  :  AutoModerator
2016-09-02 02:06:21
100  :  AutoModerator
2016-09-02 07:03:28
100  :  AutoModerator
2016-09-02 10:32:28
100  :  AutoModerator
2016-09-02 14:04:20
100  :  AutoModerator
2016-09-02 16:56:22
100  :  AutoModerator
2016-09-02 19:03:54
100  :  AutoModerator
2016-09-02 21:03:24
100  :  AutoModerator
2016-09-02 22:06:10
100  :  AutoModerator
2016-09-03 00:00:08
100  :  AutoModerator
2016-09-03 01:36:34
100  :  AutoModerator
2016-09-03 06:02:29
100  :  AutoModerator
2016-09-03 09:08:27
100  :  AutoModerator
2016-09-03 14:02:37
100  :  AutoModerator
2016-09-03 17:03:56
100  :  AutoModerator
2016-09-03 20:03:49
100  :  AutoModerator
2016-09-03 22:05:01
100  :  AutoModerator
2016-09-04 00:05:14
100  :  AutoModerator
2016-09-04 04:02:16
100  :  AutoModerator
2016-09-04 09:03:13
100  :  AutoModerator
2016-09-04 13:04:54
100  :  AutoModerator
2016-09-04 16:05:46
100  :  AutoModerator
2016-09-04 19:03:41
100  :  AutoModerator
2016-09-04 21:07:13
100  :  AutoModerator
2016-09-05 0

100  :  AutoModerator
2016-09-26 20:04:07
100  :  AutoModerator
2016-09-26 21:09:55
100  :  AutoModerator
2016-09-26 22:32:22
100  :  AutoModerator
2016-09-27 00:06:41
100  :  AutoModerator
2016-09-27 03:02:33
100  :  AutoModerator
2016-09-27 08:03:13
100  :  AutoModerator
2016-09-27 12:50:15
100  :  AutoModerator
2016-09-27 16:06:09
100  :  AutoModerator
2016-09-27 19:52:16
100  :  AutoModerator
2016-09-27 21:32:18
100  :  AutoModerator
2016-09-28 00:00:15
100  :  AutoModerator
2016-09-28 03:36:23
100  :  AutoModerator
2016-09-28 09:04:56
100  :  AutoModerator
2016-09-28 14:03:04
100  :  AutoModerator
2016-09-28 17:03:20
100  :  AutoModerator
2016-09-28 20:02:46
100  :  AutoModerator
2016-09-28 22:03:28
100  :  AutoModerator
2016-09-29 00:02:13
100  :  AutoModerator
2016-09-29 02:04:07
100  :  AutoModerator
2016-09-29 08:02:29
100  :  AutoModerator
2016-09-29 12:06:23
100  :  AutoModerator
2016-09-29 16:06:06
100  :  AutoModerator
2016-09-29 19:04:27
100  :  AutoModerator
2016-09-29 2

100  :  AutoModerator
2016-10-20 03:06:22
100  :  AutoModerator
2016-10-20 08:32:09
100  :  AutoModerator
2016-10-20 12:04:23
100  :  AutoModerator
2016-10-20 16:00:26
100  :  AutoModerator
2016-10-20 18:04:20
100  :  AutoModerator
2016-10-20 21:02:32
100  :  AutoModerator
2016-10-20 22:32:33
100  :  AutoModerator
2016-10-21 01:03:11
100  :  AutoModerator
2016-10-21 04:22:11
100  :  AutoModerator
2016-10-21 09:04:30
100  :  AutoModerator
2016-10-21 13:02:34
100  :  AutoModerator
2016-10-21 15:05:50
100  :  AutoModerator
2016-10-21 17:06:20
100  :  AutoModerator
2016-10-21 19:12:10
100  :  AutoModerator
2016-10-21 21:04:32
100  :  AutoModerator
2016-10-21 22:07:14
100  :  AutoModerator
2016-10-21 23:32:17
100  :  AutoModerator
2016-10-22 01:08:04
100  :  AutoModerator
2016-10-22 02:38:34
100  :  AutoModerator
2016-10-22 03:46:20
100  :  AutoModerator
2016-10-22 09:00:07
100  :  AutoModerator
2016-10-22 12:03:27
100  :  AutoModerator
2016-10-22 15:22:09
100  :  AutoModerator
2016-10-22 1

100  :  AutoModerator
2016-11-10 15:52:09
100  :  AutoModerator
2016-11-10 18:03:40
100  :  AutoModerator
2016-11-10 20:32:32
100  :  AutoModerator
2016-11-10 22:08:59
100  :  AutoModerator
2016-11-11 00:06:46
100  :  AutoModerator
2016-11-11 03:32:36
100  :  AutoModerator
2016-11-11 08:52:39
100  :  AutoModerator
2016-11-11 11:44:38
100  :  AutoModerator
2016-11-11 15:00:21
100  :  AutoModerator
2016-11-11 17:02:08
100  :  AutoModerator
2016-11-11 19:03:20
100  :  AutoModerator
2016-11-11 20:32:39
100  :  AutoModerator
2016-11-11 22:02:34
100  :  AutoModerator
2016-11-11 23:04:57
100  :  AutoModerator
2016-11-12 01:00:57
100  :  AutoModerator
2016-11-12 02:34:30
100  :  AutoModerator
2016-11-12 02:44:18
100  :  AutoModerator
2016-11-12 06:03:19
100  :  AutoModerator
2016-11-12 09:07:37
100  :  AutoModerator
2016-11-12 13:05:37
100  :  AutoModerator
2016-11-12 16:05:59
100  :  AutoModerator
2016-11-12 19:03:09
100  :  AutoModerator
2016-11-12 21:03:50
100  :  AutoModerator
2016-11-12 2

100  :  AutoModerator
2016-12-01 06:04:52
100  :  AutoModerator
2016-12-01 09:06:23
100  :  AutoModerator
2016-12-01 13:02:42
100  :  AutoModerator
2016-12-01 15:32:36
100  :  AutoModerator
2016-12-01 17:32:09
100  :  AutoModerator
2016-12-01 19:52:19
100  :  AutoModerator
2016-12-01 21:09:09
100  :  AutoModerator
2016-12-01 23:05:07
100  :  AutoModerator
2016-12-02 01:03:12
100  :  AutoModerator
2016-12-02 04:02:10
100  :  AutoModerator
2016-12-02 07:32:12
100  :  AutoModerator
2016-12-02 11:02:10
100  :  AutoModerator
2016-12-02 14:04:44
100  :  AutoModerator
2016-12-02 16:06:42
100  :  AutoModerator
2016-12-02 18:04:24
100  :  AutoModerator
2016-12-02 20:03:57
100  :  AutoModerator
2016-12-02 21:10:12
100  :  AutoModerator
2016-12-02 22:14:25
100  :  AutoModerator
2016-12-03 00:04:01
100  :  AutoModerator
2016-12-03 02:02:20
100  :  AutoModerator
2016-12-03 02:39:19
100  :  AutoModerator
2016-12-03 04:02:51
100  :  AutoModerator
2016-12-03 08:02:46
100  :  AutoModerator
2016-12-03 1

100  :  AutoModerator
2016-12-19 19:04:29
100  :  AutoModerator
2016-12-19 20:32:40
100  :  AutoModerator
2016-12-19 21:14:42
100  :  AutoModerator
2016-12-19 22:26:20
100  :  AutoModerator
2016-12-20 00:03:44
100  :  AutoModerator
2016-12-20 01:22:17
100  :  AutoModerator
2016-12-20 04:03:38
100  :  AutoModerator
2016-12-20 07:03:32
100  :  AutoModerator
2016-12-20 09:52:11
100  :  AutoModerator
2016-12-20 13:03:55
100  :  AutoModerator
2016-12-20 15:42:15
100  :  AutoModerator
2016-12-20 17:52:17
100  :  AutoModerator
2016-12-20 20:03:11
100  :  AutoModerator
2016-12-20 21:09:45
100  :  AutoModerator
2016-12-20 23:04:05
100  :  AutoModerator
2016-12-21 01:02:31
100  :  AutoModerator
2016-12-21 03:56:39
100  :  AutoModerator
2016-12-21 07:04:45
100  :  AutoModerator
2016-12-21 10:02:59
100  :  AutoModerator
2016-12-21 13:22:26
100  :  AutoModerator
2016-12-21 15:56:24
100  :  AutoModerator
2016-12-21 17:32:54
100  :  AutoModerator
2016-12-21 20:02:30
100  :  AutoModerator
2016-12-21 2

100  :  AutoModerator
2017-01-06 18:04:00
100  :  AutoModerator
2017-01-06 19:56:38
100  :  AutoModerator
2017-01-06 21:06:40
100  :  AutoModerator
2017-01-06 22:07:48
100  :  AutoModerator
2017-01-06 23:16:31
100  :  AutoModerator
2017-01-07 01:04:28
에러 발생
100  :  AutoModerator
2017-01-07 02:35:18
100  :  AutoModerator
2017-01-07 02:45:06
100  :  AutoModerator
2017-01-07 06:02:53
100  :  AutoModerator
2017-01-07 09:03:35
100  :  AutoModerator
2017-01-07 11:32:36
100  :  AutoModerator
2017-01-07 14:22:17
100  :  AutoModerator
2017-01-07 17:03:40
100  :  AutoModerator
2017-01-07 19:04:39
100  :  AutoModerator
2017-01-07 21:03:44
100  :  AutoModerator
2017-01-07 22:32:16
100  :  AutoModerator
2017-01-08 00:06:59
100  :  AutoModerator
2017-01-08 03:04:06
100  :  AutoModerator
2017-01-08 06:16:52
100  :  AutoModerator
2017-01-08 09:05:29
100  :  AutoModerator
2017-01-08 12:04:37
100  :  AutoModerator
2017-01-08 15:02:52
100  :  AutoModerator
2017-01-08 16:44:10
100  :  AutoModerator
2017-0

100  :  AutoModerator
2017-01-23 21:08:19
100  :  AutoModerator
2017-01-23 22:06:05
100  :  AutoModerator
2017-01-23 23:12:57
100  :  AutoModerator
2017-01-24 01:02:33
100  :  AutoModerator
2017-01-24 02:50:18
100  :  AutoModerator
2017-01-24 06:03:32
100  :  AutoModerator
2017-01-24 09:03:27
100  :  AutoModerator
2017-01-24 11:24:39
100  :  AutoModerator
2017-01-24 14:06:22
100  :  AutoModerator
2017-01-24 16:42:09
100  :  AutoModerator
2017-01-24 18:34:31
100  :  AutoModerator
2017-01-24 20:12:12
100  :  AutoModerator
2017-01-24 22:03:08
100  :  AutoModerator
2017-01-24 23:12:10
100  :  AutoModerator
2017-01-25 01:32:32
100  :  AutoModerator
2017-01-25 04:42:12
100  :  AutoModerator
2017-01-25 08:04:27
100  :  AutoModerator
2017-01-25 10:28:13
100  :  AutoModerator
2017-01-25 13:46:33
100  :  AutoModerator
2017-01-25 16:02:18
100  :  AutoModerator
2017-01-25 17:33:14
100  :  AutoModerator
2017-01-25 19:42:09
100  :  AutoModerator
2017-01-25 21:07:31
100  :  AutoModerator
2017-01-25 2

100  :  AutoModerator
2017-02-09 23:05:54
100  :  AutoModerator
2017-02-10 01:06:26
100  :  AutoModerator
2017-02-10 04:04:16
100  :  AutoModerator
2017-02-10 07:32:12
100  :  AutoModerator
2017-02-10 09:24:32
100  :  AutoModerator
2017-02-10 13:02:38
100  :  AutoModerator
2017-02-10 15:03:51
100  :  AutoModerator
2017-02-10 16:42:16
100  :  AutoModerator
2017-02-10 18:05:52
100  :  AutoModerator
2017-02-10 20:02:15
100  :  AutoModerator
2017-02-10 21:05:48
100  :  AutoModerator
2017-02-10 22:06:20
100  :  AutoModerator
2017-02-10 23:07:58
100  :  AutoModerator
2017-02-11 01:02:15
100  :  AutoModerator
2017-02-11 02:33:09
100  :  AutoModerator
2017-02-11 02:42:51
100  :  AutoModerator
2017-02-11 05:02:59
100  :  AutoModerator
2017-02-11 08:04:19
100  :  AutoModerator
2017-02-11 10:04:40
100  :  AutoModerator
2017-02-11 13:06:12
100  :  AutoModerator
2017-02-11 16:02:53
100  :  AutoModerator
2017-02-11 18:02:10
100  :  AutoModerator
2017-02-11 20:02:37
100  :  AutoModerator
2017-02-11 2

100  :  AutoModerator
2017-02-27 02:14:57
100  :  AutoModerator
2017-02-27 05:07:53
100  :  AutoModerator
2017-02-27 08:03:35
100  :  AutoModerator
2017-02-27 09:13:17
100  :  AutoModerator
2017-02-27 12:03:33
100  :  AutoModerator
2017-02-27 14:13:17
100  :  AutoModerator
2017-02-27 16:04:07
100  :  AutoModerator
2017-02-27 17:11:28
100  :  AutoModerator
2017-02-27 19:06:56
100  :  AutoModerator
2017-02-27 20:30:05
100  :  AutoModerator
2017-02-27 21:34:33
100  :  AutoModerator
2017-02-27 22:39:40
100  :  AutoModerator
2017-02-28 00:17:21
100  :  AutoModerator
2017-02-28 02:08:46
100  :  AutoModerator
2017-02-28 05:12:49
100  :  AutoModerator
2017-02-28 09:03:55
100  :  AutoModerator
2017-02-28 11:35:21
100  :  AutoModerator
2017-02-28 15:07:03
100  :  AutoModerator
2017-02-28 17:16:26
100  :  AutoModerator
2017-02-28 19:12:38
100  :  AutoModerator
2017-02-28 21:10:17
100  :  AutoModerator
2017-02-28 22:16:26
100  :  AutoModerator
2017-03-01 00:10:25
100  :  AutoModerator
2017-03-01 0

100  :  AutoModerator
2017-03-15 08:16:58
100  :  AutoModerator
2017-03-15 11:02:32
100  :  AutoModerator
2017-03-15 14:03:41
100  :  AutoModerator
2017-03-15 16:04:15
100  :  AutoModerator
2017-03-15 17:42:12
100  :  AutoModerator
2017-03-15 19:26:42
100  :  AutoModerator
2017-03-15 21:03:56
100  :  AutoModerator
2017-03-15 22:08:51
100  :  AutoModerator
2017-03-15 23:33:36
100  :  AutoModerator
2017-03-16 01:06:43
100  :  AutoModerator
2017-03-16 03:22:42
100  :  AutoModerator
2017-03-16 06:32:57
100  :  AutoModerator
2017-03-16 09:06:39
100  :  AutoModerator
2017-03-16 12:14:37
100  :  AutoModerator
2017-03-16 15:03:56
100  :  AutoModerator
2017-03-16 17:05:09
100  :  AutoModerator
2017-03-16 19:05:28
100  :  AutoModerator
2017-03-16 21:00:43
100  :  AutoModerator
2017-03-16 22:07:38
100  :  AutoModerator
2017-03-17 00:02:07
100  :  AutoModerator
2017-03-17 02:02:15
100  :  AutoModerator
2017-03-17 05:00:46
100  :  AutoModerator
2017-03-17 08:03:06
100  :  AutoModerator
2017-03-17 1

100  :  AutoModerator
2017-03-31 16:04:45
100  :  AutoModerator
2017-03-31 17:14:49
100  :  AutoModerator
2017-03-31 19:05:52
100  :  AutoModerator
2017-03-31 20:10:33
100  :  AutoModerator
2017-03-31 21:13:50
100  :  AutoModerator
2017-03-31 22:12:38
100  :  AutoModerator
2017-03-31 23:24:51
100  :  AutoModerator
2017-04-01 00:52:12
100  :  AutoModerator
2017-04-01 02:06:40
100  :  AutoModerator
2017-04-01 02:41:14
100  :  AutoModerator
2017-04-01 04:04:53
100  :  AutoModerator
2017-04-01 06:58:35
100  :  AutoModerator
2017-04-01 09:04:46
100  :  AutoModerator
2017-04-01 11:02:29
100  :  AutoModerator
2017-04-01 13:06:18
100  :  AutoModerator
2017-04-01 15:03:26
100  :  AutoModerator
2017-04-01 16:22:10
100  :  AutoModerator
2017-04-01 18:04:16
100  :  AutoModerator
2017-04-01 19:42:09
100  :  AutoModerator
2017-04-01 21:03:58
100  :  AutoModerator
2017-04-01 22:08:53
100  :  AutoModerator
2017-04-01 23:09:36
100  :  AutoModerator
2017-04-02 01:00:44
100  :  AutoModerator
2017-04-02 0

100  :  AutoModerator
2017-04-15 12:12:24
100  :  AutoModerator
2017-04-15 14:32:31
100  :  AutoModerator
2017-04-15 16:32:36
100  :  AutoModerator
2017-04-15 18:05:41
100  :  AutoModerator
2017-04-15 20:02:18
100  :  AutoModerator
2017-04-15 21:07:34
100  :  AutoModerator
2017-04-15 22:32:44
100  :  AutoModerator
2017-04-16 00:05:21
100  :  AutoModerator
2017-04-16 02:04:05
100  :  AutoModerator
2017-04-16 04:46:43
100  :  AutoModerator
2017-04-16 07:12:17
100  :  AutoModerator
2017-04-16 09:30:27
100  :  AutoModerator
2017-04-16 12:32:56
100  :  AutoModerator
2017-04-16 14:34:50
100  :  AutoModerator
2017-04-16 16:11:39
100  :  AutoModerator
2017-04-16 18:04:27
100  :  AutoModerator
2017-04-16 19:24:42
100  :  AutoModerator
2017-04-16 21:04:47
100  :  AutoModerator
2017-04-16 22:08:34
100  :  AutoModerator
2017-04-16 23:46:30
100  :  AutoModerator
2017-04-17 01:09:20
100  :  AutoModerator
2017-04-17 03:04:50
100  :  AutoModerator
2017-04-17 05:42:10
100  :  AutoModerator
2017-04-17 0

100  :  AutoModerator
2017-04-30 19:03:01
100  :  AutoModerator
2017-04-30 20:08:54
100  :  AutoModerator
2017-04-30 21:32:17
100  :  AutoModerator
2017-04-30 23:04:30
100  :  AutoModerator
2017-05-01 00:08:58
100  :  AutoModerator
2017-05-01 02:03:52
100  :  AutoModerator
2017-05-01 04:05:12
100  :  AutoModerator
2017-05-01 06:48:19
100  :  AutoModerator
2017-05-01 09:04:07
100  :  AutoModerator
2017-05-01 11:03:05
100  :  AutoModerator
2017-05-01 13:07:08
100  :  AutoModerator
2017-05-01 15:01:02
100  :  AutoModerator
2017-05-01 16:05:31
100  :  AutoModerator
2017-05-01 17:07:17
100  :  AutoModerator
2017-05-01 18:08:29
100  :  AutoModerator
2017-05-01 19:26:49
100  :  AutoModerator
2017-05-01 21:00:34
100  :  AutoModerator
2017-05-01 21:46:32
100  :  AutoModerator
2017-05-01 22:33:35
100  :  AutoModerator
2017-05-02 00:01:04
100  :  AutoModerator
2017-05-02 01:05:56
100  :  AutoModerator
2017-05-02 02:33:52
100  :  AutoModerator
2017-05-02 05:00:55
100  :  AutoModerator
2017-05-02 0

100  :  AutoModerator
2017-05-15 16:02:16
100  :  AutoModerator
2017-05-15 17:06:41
100  :  AutoModerator
2017-05-15 18:12:15
100  :  AutoModerator
2017-05-15 19:42:07
100  :  AutoModerator
2017-05-15 21:02:15
100  :  AutoModerator
2017-05-15 22:00:52
100  :  AutoModerator
2017-05-15 23:03:26
100  :  AutoModerator
2017-05-16 00:05:51
100  :  AutoModerator
2017-05-16 01:46:56
100  :  AutoModerator
2017-05-16 04:02:58
100  :  AutoModerator
2017-05-16 06:32:26
100  :  AutoModerator
2017-05-16 09:05:26
100  :  AutoModerator
2017-05-16 12:02:11
100  :  AutoModerator
2017-05-16 14:06:19
100  :  AutoModerator
2017-05-16 16:06:59
100  :  AutoModerator
2017-05-16 18:03:55
100  :  AutoModerator
2017-05-16 19:42:16
100  :  AutoModerator
2017-05-16 21:06:21
100  :  AutoModerator
2017-05-16 22:32:16
100  :  AutoModerator
2017-05-17 00:04:24
100  :  AutoModerator
2017-05-17 02:04:16
100  :  AutoModerator
2017-05-17 05:00:40
100  :  AutoModerator
2017-05-17 08:02:11
100  :  AutoModerator
2017-05-17 1

100  :  AutoModerator
2017-05-30 17:08:06
100  :  AutoModerator
2017-05-30 19:05:06
100  :  AutoModerator
2017-05-30 20:08:44
100  :  AutoModerator
2017-05-30 21:16:11
100  :  AutoModerator
2017-05-30 23:06:41
100  :  AutoModerator
2017-05-31 00:11:12
100  :  AutoModerator
2017-05-31 02:20:50
100  :  AutoModerator
2017-05-31 05:20:48
100  :  AutoModerator
2017-05-31 08:15:14
100  :  AutoModerator
2017-05-31 10:25:55
100  :  AutoModerator
2017-05-31 13:12:57
100  :  AutoModerator
2017-05-31 15:11:41
100  :  AutoModerator
2017-05-31 17:07:36
100  :  AutoModerator
2017-05-31 18:35:59
100  :  AutoModerator
2017-05-31 20:08:16
100  :  AutoModerator
2017-05-31 21:12:02
100  :  AutoModerator
2017-05-31 22:35:21
100  :  AutoModerator
2017-06-01 00:05:49
100  :  AutoModerator
2017-06-01 01:15:07
100  :  AutoModerator
2017-06-01 04:05:32
100  :  AutoModerator
2017-06-01 06:35:43
100  :  AutoModerator
2017-06-01 09:06:57
100  :  AutoModerator
2017-06-01 11:07:34
100  :  AutoModerator
2017-06-01 1

100  :  AutoModerator
2017-06-16 21:07:39
100  :  AutoModerator
2017-06-16 22:10:10
100  :  AutoModerator
2017-06-16 23:12:30
100  :  AutoModerator
2017-06-17 01:06:01
100  :  AutoModerator
2017-06-17 02:37:40
100  :  AutoModerator
2017-06-17 02:47:40
100  :  AutoModerator
2017-06-17 06:07:48
100  :  AutoModerator
2017-06-17 09:08:59
100  :  AutoModerator
2017-06-17 12:35:52
100  :  AutoModerator
2017-06-17 15:09:12
100  :  AutoModerator
2017-06-17 17:11:17
100  :  AutoModerator
2017-06-17 19:11:14
100  :  AutoModerator
2017-06-17 21:06:34
100  :  AutoModerator
2017-06-17 23:01:10
100  :  AutoModerator
2017-06-18 00:30:50
100  :  AutoModerator
2017-06-18 04:08:06
100  :  AutoModerator
2017-06-18 08:15:47
100  :  AutoModerator
2017-06-18 11:07:25
100  :  AutoModerator
2017-06-18 14:10:08
100  :  AutoModerator
2017-06-18 16:11:02
100  :  AutoModerator
2017-06-18 18:09:43
100  :  AutoModerator
2017-06-18 20:08:52
100  :  AutoModerator
2017-06-18 21:15:55
100  :  AutoModerator
2017-06-18 2

100  :  AutoModerator
2017-07-04 23:09:01
100  :  AutoModerator
2017-07-05 01:08:39
100  :  AutoModerator
2017-07-05 05:10:59
100  :  AutoModerator
2017-07-05 09:08:16
100  :  AutoModerator
2017-07-05 13:05:48
100  :  AutoModerator
2017-07-05 15:10:43
100  :  AutoModerator
2017-07-05 17:11:14
100  :  AutoModerator
2017-07-05 19:10:56
100  :  AutoModerator
2017-07-05 21:06:11
100  :  AutoModerator
2017-07-05 22:10:59
100  :  AutoModerator
2017-07-05 23:36:15
100  :  AutoModerator
2017-07-06 02:00:28
100  :  AutoModerator
2017-07-06 05:35:41
100  :  AutoModerator
2017-07-06 09:09:47
100  :  AutoModerator
2017-07-06 13:16:04
100  :  AutoModerator
2017-07-06 16:09:44
100  :  AutoModerator
2017-07-06 18:10:54
100  :  AutoModerator
2017-07-06 20:10:52
100  :  AutoModerator
2017-07-06 22:07:27
100  :  AutoModerator
2017-07-07 00:05:14
100  :  AutoModerator
2017-07-07 02:07:53
100  :  AutoModerator
2017-07-07 07:06:08
100  :  AutoModerator
2017-07-07 09:13:27
100  :  AutoModerator
2017-07-07 1

100  :  AutoModerator
2017-07-23 18:15:26
100  :  AutoModerator
2017-07-23 20:10:17
100  :  AutoModerator
2017-07-23 22:05:29
100  :  AutoModerator
2017-07-23 23:31:01
100  :  AutoModerator
2017-07-24 01:10:41
100  :  AutoModerator
2017-07-24 04:41:01
100  :  AutoModerator
2017-07-24 08:08:11
100  :  AutoModerator
2017-07-24 09:16:52
100  :  AutoModerator
2017-07-24 12:45:39
100  :  AutoModerator
2017-07-24 15:05:31
100  :  AutoModerator
2017-07-24 16:10:21
100  :  AutoModerator
2017-07-24 17:13:25
100  :  AutoModerator
2017-07-24 19:07:04
100  :  AutoModerator
2017-07-24 20:08:40
100  :  AutoModerator
2017-07-24 21:10:51
100  :  AutoModerator
2017-07-24 22:10:09
100  :  AutoModerator
2017-07-24 23:12:33
100  :  AutoModerator
2017-07-25 00:40:09
100  :  AutoModerator
2017-07-25 03:05:12
100  :  AutoModerator
2017-07-25 07:05:33
100  :  AutoModerator
2017-07-25 09:35:55
100  :  AutoModerator
2017-07-25 13:09:15
100  :  AutoModerator
2017-07-25 16:07:07
100  :  AutoModerator
2017-07-25 1

100  :  AutoModerator
2017-08-11 05:07:43
100  :  AutoModerator
2017-08-11 09:08:27
100  :  AutoModerator
2017-08-11 12:07:28
100  :  AutoModerator
2017-08-11 15:05:07
100  :  AutoModerator
2017-08-11 16:13:27
100  :  AutoModerator
2017-08-11 18:07:41
100  :  AutoModerator
2017-08-11 19:20:53
100  :  AutoModerator
2017-08-11 21:05:40
100  :  AutoModerator
2017-08-11 22:07:50
100  :  AutoModerator
2017-08-11 23:09:33
100  :  AutoModerator
2017-08-12 00:13:29
100  :  AutoModerator
2017-08-12 02:36:09
100  :  AutoModerator
2017-08-12 02:46:03
100  :  AutoModerator
2017-08-12 06:05:57
100  :  AutoModerator
2017-08-12 09:07:17
100  :  AutoModerator
2017-08-12 12:35:58
100  :  AutoModerator
2017-08-12 15:09:02
100  :  AutoModerator
2017-08-12 17:11:10
100  :  AutoModerator
2017-08-12 19:15:25
100  :  AutoModerator
2017-08-12 21:07:18
100  :  AutoModerator
2017-08-12 23:07:03
100  :  AutoModerator
2017-08-13 01:07:40
100  :  AutoModerator
2017-08-13 05:07:35
100  :  AutoModerator
2017-08-13 0

100  :  AutoModerator
2017-08-29 15:08:12
100  :  AutoModerator
2017-08-29 17:10:27
100  :  AutoModerator
2017-08-29 19:09:34
100  :  AutoModerator
2017-08-29 21:05:46
100  :  AutoModerator
2017-08-29 22:12:24
100  :  AutoModerator
2017-08-30 00:08:03
100  :  AutoModerator
2017-08-30 03:07:06
100  :  AutoModerator
2017-08-30 07:50:22
100  :  AutoModerator
2017-08-30 11:06:08
100  :  AutoModerator
2017-08-30 14:09:31
100  :  AutoModerator
2017-08-30 16:21:10
100  :  AutoModerator
2017-08-30 18:10:46
100  :  AutoModerator
2017-08-30 20:06:53
100  :  AutoModerator
2017-08-30 21:12:51
100  :  AutoModerator
2017-08-30 23:07:02
100  :  AutoModerator
2017-08-31 00:11:39
100  :  AutoModerator
2017-08-31 03:06:09
100  :  AutoModerator
2017-08-31 07:08:30
100  :  AutoModerator
2017-08-31 11:05:14
100  :  AutoModerator
2017-08-31 15:05:53
100  :  AutoModerator
2017-08-31 17:08:56
100  :  AutoModerator
2017-08-31 19:08:14
100  :  AutoModerator
2017-08-31 21:05:56
100  :  AutoModerator
2017-08-31 2

100  :  AutoModerator
2017-09-16 10:06:01
100  :  AutoModerator
2017-09-16 14:05:41
100  :  AutoModerator
2017-09-16 16:09:12
100  :  AutoModerator
2017-09-16 18:09:10
100  :  AutoModerator
2017-09-16 20:08:37
100  :  AutoModerator
2017-09-16 22:06:28
100  :  AutoModerator
2017-09-16 23:15:08
100  :  AutoModerator
2017-09-17 02:00:15
100  :  AutoModerator
2017-09-17 06:07:58
100  :  AutoModerator
2017-09-17 09:11:41
100  :  AutoModerator
2017-09-17 13:06:06
100  :  AutoModerator
2017-09-17 15:09:42
100  :  AutoModerator
2017-09-17 17:08:19
100  :  AutoModerator
2017-09-17 19:05:37
100  :  AutoModerator
2017-09-17 20:11:28
100  :  AutoModerator
2017-09-17 22:07:03
100  :  AutoModerator
2017-09-17 23:36:28
100  :  AutoModerator
2017-09-18 01:09:42
100  :  AutoModerator
2017-09-18 04:07:47
100  :  AutoModerator
2017-09-18 08:06:41
100  :  AutoModerator
2017-09-18 09:15:36
100  :  AutoModerator
2017-09-18 12:35:42
100  :  AutoModerator
2017-09-18 14:11:46
100  :  AutoModerator
2017-09-18 1

100  :  AutoModerator
2017-10-03 09:05:25
100  :  AutoModerator
2017-10-03 11:08:01
100  :  AutoModerator
2017-10-03 14:06:29
100  :  AutoModerator
2017-10-03 16:07:32
100  :  AutoModerator
2017-10-03 17:37:08
100  :  AutoModerator
2017-10-03 19:10:18
100  :  AutoModerator
2017-10-03 21:06:41
100  :  AutoModerator
2017-10-03 22:10:19
100  :  AutoModerator
2017-10-03 23:40:16
100  :  AutoModerator
2017-10-04 01:20:49
100  :  AutoModerator
2017-10-04 05:05:28
100  :  AutoModerator
2017-10-04 08:09:13
100  :  AutoModerator
2017-10-04 10:25:15
100  :  AutoModerator
2017-10-04 13:10:22
100  :  AutoModerator
2017-10-04 15:11:27
100  :  AutoModerator
2017-10-04 17:08:32
100  :  AutoModerator
2017-10-04 19:05:22
100  :  AutoModerator
2017-10-04 20:11:15
100  :  AutoModerator
2017-10-04 21:55:54
100  :  AutoModerator
2017-10-04 23:08:05
100  :  AutoModerator
2017-10-05 00:15:04
100  :  AutoModerator
2017-10-05 02:21:00
100  :  AutoModerator
2017-10-05 06:07:45
100  :  AutoModerator
2017-10-05 0

100  :  AutoModerator
2017-10-18 21:35:49
100  :  AutoModerator
2017-10-18 23:05:46
100  :  AutoModerator
2017-10-19 00:09:53
100  :  AutoModerator
2017-10-19 02:06:14
100  :  AutoModerator
2017-10-19 05:08:06
100  :  AutoModerator
2017-10-19 08:16:32
100  :  AutoModerator
2017-10-19 10:55:42
100  :  AutoModerator
2017-10-19 14:05:11
100  :  AutoModerator
2017-10-19 16:07:30
100  :  AutoModerator
2017-10-19 17:35:50
100  :  AutoModerator
2017-10-19 19:10:12
100  :  AutoModerator
2017-10-19 21:06:14
100  :  AutoModerator
2017-10-19 22:12:12
100  :  AutoModerator
2017-10-20 00:06:42
100  :  AutoModerator
2017-10-20 02:05:48
100  :  AutoModerator
2017-10-20 05:40:11
100  :  AutoModerator
2017-10-20 08:56:07
100  :  AutoModerator
2017-10-20 10:15:08
100  :  AutoModerator
2017-10-20 13:07:20
100  :  AutoModerator
2017-10-20 14:55:12
100  :  AutoModerator
2017-10-20 16:09:39
100  :  AutoModerator
2017-10-20 17:14:07
100  :  AutoModerator
2017-10-20 19:06:02
100  :  AutoModerator
2017-10-20 2

100  :  AutoModerator
2017-11-03 08:08:53
100  :  AutoModerator
2017-11-03 10:09:02
100  :  AutoModerator
2017-11-03 13:09:42
100  :  AutoModerator
2017-11-03 15:07:20
100  :  AutoModerator
2017-11-03 16:13:16
100  :  AutoModerator
2017-11-03 18:00:33
100  :  AutoModerator
2017-11-03 19:09:48
100  :  AutoModerator
2017-11-03 20:11:14
100  :  AutoModerator
2017-11-03 21:13:09
100  :  AutoModerator
2017-11-03 22:13:37
100  :  AutoModerator
2017-11-03 23:16:11
100  :  AutoModerator
2017-11-04 01:05:12
100  :  AutoModerator
2017-11-04 02:10:11
100  :  AutoModerator
2017-11-04 02:44:43
100  :  AutoModerator
2017-11-04 04:08:34
100  :  AutoModerator
2017-11-04 07:08:12
100  :  AutoModerator
2017-11-04 09:12:09
100  :  AutoModerator
2017-11-04 12:35:50
100  :  AutoModerator
2017-11-04 14:45:57
100  :  AutoModerator
2017-11-04 16:36:06
100  :  AutoModerator
2017-11-04 18:09:05
100  :  AutoModerator
2017-11-04 20:05:46
100  :  AutoModerator
2017-11-04 21:11:01
100  :  AutoModerator
2017-11-04 2

100  :  AutoModerator
2017-11-18 06:05:58
100  :  AutoModerator
2017-11-18 08:46:08
100  :  AutoModerator
2017-11-18 10:25:14
100  :  AutoModerator
2017-11-18 13:10:56
100  :  AutoModerator
2017-11-18 15:11:51
100  :  AutoModerator
2017-11-18 17:10:56
100  :  AutoModerator
2017-11-18 19:06:14
100  :  AutoModerator
2017-11-18 20:11:51
100  :  AutoModerator
2017-11-18 22:06:27
100  :  AutoModerator
2017-11-18 23:11:26
100  :  AutoModerator
2017-11-19 01:09:11
100  :  AutoModerator
2017-11-19 04:05:26
100  :  AutoModerator
2017-11-19 07:08:02
100  :  AutoModerator
2017-11-19 09:12:49
100  :  AutoModerator
2017-11-19 12:20:59
100  :  AutoModerator
2017-11-19 14:16:16
100  :  AutoModerator
2017-11-19 16:08:39
100  :  AutoModerator
2017-11-19 17:16:05
100  :  AutoModerator
2017-11-19 19:08:39
100  :  AutoModerator
2017-11-19 20:30:24
100  :  AutoModerator
2017-11-19 22:05:53
100  :  AutoModerator
2017-11-19 23:10:35
100  :  AutoModerator
2017-11-20 01:05:14
100  :  AutoModerator
2017-11-20 0